<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>zfBrainAsymmetry - The parapineal ‘cell type’ - Part 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">zfBrainAsymmetry</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./session.html" rel="" target="">
 <span class="menu-text">Session</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-aims" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Aims</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-aims">    
        <li>
    <a class="dropdown-item" href="./Aim1_part1.html" rel="" target="">
 <span class="dropdown-text">The parapineal ‘cell type’ - Part 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Aim1_part2.html" rel="" target="">
 <span class="dropdown-text">The parapineal ‘cell type’ - Part 2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Aim2.html" rel="" target="">
 <span class="dropdown-text">The sox1a ‘phenotype’</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#identification-of-epiphysis-cells" id="toc-identification-of-epiphysis-cells" class="nav-link active" data-scroll-target="#identification-of-epiphysis-cells"><span class="header-section-number">1</span> Identification of epiphysis cells</a>
  <ul class="collapse">
  <li><a href="#premise" id="toc-premise" class="nav-link" data-scroll-target="#premise"><span class="header-section-number">1.1</span> <strong>Premise</strong></a></li>
  <li><a href="#experimental-design" id="toc-experimental-design" class="nav-link" data-scroll-target="#experimental-design"><span class="header-section-number">1.2</span> Experimental design</a></li>
  <li><a href="#initialise-r-environment" id="toc-initialise-r-environment" class="nav-link" data-scroll-target="#initialise-r-environment"><span class="header-section-number">1.3</span> Initialise R environment</a></li>
  </ul></li>
  <li><a href="#merscope-panel-design" id="toc-merscope-panel-design" class="nav-link" data-scroll-target="#merscope-panel-design"><span class="header-section-number">2</span> MERSCOPE panel design</a>
  <ul class="collapse">
  <li><a href="#call-markers" id="toc-call-markers" class="nav-link" data-scroll-target="#call-markers"><span class="header-section-number">2.1</span> Call markers</a></li>
  <li><a href="#cross-reference-with-zfin" id="toc-cross-reference-with-zfin" class="nav-link" data-scroll-target="#cross-reference-with-zfin"><span class="header-section-number">2.2</span> Cross-reference with ZFIN</a></li>
  <li><a href="#gene-panel-for-merscope---final" id="toc-gene-panel-for-merscope---final" class="nav-link" data-scroll-target="#gene-panel-for-merscope---final"><span class="header-section-number">2.3</span> 140 Gene Panel for MERSCOPE - final</a></li>
  <li><a href="#additional-insights" id="toc-additional-insights" class="nav-link" data-scroll-target="#additional-insights"><span class="header-section-number">2.4</span> Additional insights</a></li>
  </ul></li>
  <li><a href="#pre-processing-of-merscope-data" id="toc-pre-processing-of-merscope-data" class="nav-link" data-scroll-target="#pre-processing-of-merscope-data"><span class="header-section-number">3</span> Pre-processing of MERSCOPE data</a>
  <ul class="collapse">
  <li><a href="#cell-segmentation" id="toc-cell-segmentation" class="nav-link" data-scroll-target="#cell-segmentation"><span class="header-section-number">3.1</span> Cell segmentation</a></li>
  </ul></li>
  <li><a href="#clustering-of-merscope-data" id="toc-clustering-of-merscope-data" class="nav-link" data-scroll-target="#clustering-of-merscope-data"><span class="header-section-number">4</span> Clustering of MERSCOPE data</a></li>
  <li><a href="#summary-notes-for-aim-1---part-2" id="toc-summary-notes-for-aim-1---part-2" class="nav-link" data-scroll-target="#summary-notes-for-aim-1---part-2"><span class="header-section-number">5</span> Summary notes for Aim 1 - Part 2</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">6</span> References</a>
  <ul class="collapse">
  <li><a href="#packages" id="toc-packages" class="nav-link" data-scroll-target="#packages"><span class="header-section-number">6.1</span> Packages</a></li>
  <li><a href="#previous-work" id="toc-previous-work" class="nav-link" data-scroll-target="#previous-work"><span class="header-section-number">6.2</span> Previous work</a></li>
  </ul></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="header-section-number">7</span> Session Info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">The parapineal ‘cell type’ - Part 2</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="identification-of-epiphysis-cells" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="identification-of-epiphysis-cells"><span class="header-section-number">1</span> Identification of epiphysis cells</h2>
<section id="premise" class="level3" data-number="1.1">
<h3 data-number="1.1" class="anchored" data-anchor-id="premise"><span class="header-section-number">1.1</span> <strong>Premise</strong></h3>
<p>In the previous <a href="./Aim1_part1.html">section</a>, two putative epiphysis cell clusters were identified in a ~30-32 hpf scRNA-seq dataset. Can this result be replicated in the embryo? In other words, how likely are the inferred clusters to be from the epiphysis?</p>
</section>
<section id="experimental-design" class="level3" data-number="1.2">
<h3 data-number="1.2" class="anchored" data-anchor-id="experimental-design"><span class="header-section-number">1.2</span> Experimental design</h3>
<p>MERFISH (<strong>M</strong>ultiplexed <strong>E</strong>rror-<strong>R</strong>obust <strong>F</strong>luorescence <strong>i</strong>n <strong>s</strong>itu <strong>H</strong>ybridisation) is an imaging-based approach capable of measuring RNA transcripts at single cell resolution. Unlike the destructive nature of scRNA-seq, <em>in vivo</em> spatial information (where each measured transcript is located in the sample) is preserved. <a href="https://vizgen.com/technology/">Vizgen MERSCOPE</a> offers a commercial platform to perform MERFISH and was employed for this study.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/merscope-experiment.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Schematic of experiment design. A detailed sample preparation protocol can be found on <a href="https://vchongmorrison.github.io/protocols/" class="uri">https://vchongmorrison.github.io/protocols/</a></figcaption>
</figure>
</div>
<p>As illustrated above, only x2 10 micron tissue sections were collected for MERFISH. This was restricted by the maximum imaging area on the MERSCOPE slide. Therefore up to 5 embryos were simultaneously mounted in a tissue block, with the caveat that there will be unavoidable inter-embryo variation of the resulting tissue slices. Depending on how the embryo was mounted, the first (and second) section could be from the parapineal/pineal region (<em>sox1a</em> / <em>bsx</em> ++), or 1 or 2 planes later towards the pineal region only (<em>bsx</em> +).</p>
<p>The final dataset consists of x2 sections each of x2 biological replicates/embryos (wild type, ~32-36 hpf). For sake of brevity, each section is named ‘anterior’ or ‘posterior’ depending on their relative location within the embryo they originated from.</p>
</section>
<section id="initialise-r-environment" class="level3" data-number="1.3">
<h3 data-number="1.3" class="anchored" data-anchor-id="initialise-r-environment"><span class="header-section-number">1.3</span> Initialise R environment</h3>
<p><a href="./session.html">Session</a></p>
</section>
</section>
<section id="merscope-panel-design" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="merscope-panel-design"><span class="header-section-number">2</span> MERSCOPE panel design</h2>
<p>Vizgen offers the option to build custom gene panels of up to 140, 300, 500 or 1000 genes at increasingly extortionate prices. Since the aim of the MERFISH experiment was to query the two putative scRNA-seq epiphysis clusters <code>2</code> and <code>19</code> from the previously analysed <code>ss_e107</code> <code>SeuratObject</code>, a 140 gene panel was designed based on this result.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"../foxd3flh-5k_ss-e107.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes that are well known to be expressed in the epiphysis</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"bsx"</span>, <span class="st">"noto"</span>, <span class="st">"otx5"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoPlotVarsAcrossGroups</span>(ss_e107, markers, <span class="at">group.by =</span> <span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">vlnplot.lineweight =</span> <span class="fl">0.2</span>, <span class="at">jitter.size =</span> <span class="fl">0.3</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">main =</span> <span class="st">"ss_e107 (mean of bsx + noto + otx5)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part2_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="call-markers" class="level3" data-number="2.1">
<h3 data-number="2.1" class="anchored" data-anchor-id="call-markers"><span class="header-section-number">2.1</span> Call markers</h3>
<p>There is an overwhelming number of methods in the scientific literature for scRNA-seq marker identification. Depending on your goal(s) for performing marker identification, some approaches might be better suited than others.</p>
<p>In this case, the main goal is to shortlist genes that could be MERFISH-measured in the embryo, including (hopefully) the epiphysis region. As a first pass, we run <code>FindMarkers</code> (implemented in <code>Seurat</code>) on clusters <code>2</code> and <code>19</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>cl2.de.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(ss_e107, <span class="at">ident.1 =</span> <span class="st">"2"</span>, <span class="at">ident.2 =</span> <span class="cn">NULL</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>, <span class="at">assay =</span> <span class="st">"RNA"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(cl2.de.markers)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6936</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add new column of gene IDs called 'markers'</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>cl2.de.markers<span class="sc">$</span>markers <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cl2.de.markers)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cl2.de.markers, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 p_val avg_log2FC pct.1 pct.2 p_val_adj          markers
otx1                 0   2.755492 0.938 0.100         0             otx1
otx2b                0   3.250256 0.810 0.044         0            otx2b
barhl2               0   3.424283 0.812 0.074         0           barhl2
si:ch211-69b22.5     0   2.500534 0.808 0.070         0 si:ch211-69b22.5
otx2a                0   2.263520 0.821 0.097         0            otx2a
wnt4                 0   2.333122 0.783 0.076         0             wnt4
fabp11a              0   4.230501 0.968 0.270         0          fabp11a
zic4                 0   2.272131 0.778 0.096         0             zic4
sall3b               0   2.044002 0.670 0.055         0           sall3b
tpbga                0   3.059875 0.655 0.045         0            tpbga</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>cl19.de.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(ss_e107, <span class="at">ident.1 =</span> <span class="st">"19"</span>, <span class="at">ident.2 =</span> <span class="cn">NULL</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>, <span class="at">assay =</span> <span class="st">"RNA"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(cl19.de.markers)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 11487</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add new column of gene IDs called 'markers'</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>cl19.de.markers<span class="sc">$</span>markers <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cl19.de.markers)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cl19.de.markers, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                          p_val avg_log2FC pct.1 pct.2     p_val_adj
bsx                0.000000e+00   5.801700 0.910 0.030  0.000000e+00
isl1               0.000000e+00   5.476998 0.888 0.031  0.000000e+00
six3a              0.000000e+00   6.449455 0.708 0.014  0.000000e+00
layna              0.000000e+00   6.930195 0.663 0.011  0.000000e+00
si:ch211-67e16.11  0.000000e+00   5.451577 0.663 0.014  0.000000e+00
zgc:165481        8.312723e-294   6.072816 0.629 0.013 1.990565e-289
LHX3              2.098162e-279   8.568871 0.326 0.001 5.024258e-275
btbd3b            6.471199e-274   4.982485 0.573 0.011 1.549593e-269
nxph1             3.397695e-248   6.210768 0.742 0.027 8.136121e-244
skor1b            1.170595e-245   6.371964 0.618 0.017 2.803108e-241
                            markers
bsx                             bsx
isl1                           isl1
six3a                         six3a
layna                         layna
si:ch211-67e16.11 si:ch211-67e16.11
zgc:165481               zgc:165481
LHX3                           LHX3
btbd3b                       btbd3b
nxph1                         nxph1
skor1b                       skor1b</code></pre>
</div>
</div>
<p><strong>Using the default cut-offs resulted in ~7 and 11k of markers.</strong> Clearly these need to be significantly narrowed down to build a 140 gene panel.</p>
</section>
<section id="cross-reference-with-zfin" class="level3" data-number="2.2">
<h3 data-number="2.2" class="anchored" data-anchor-id="cross-reference-with-zfin"><span class="header-section-number">2.2</span> Cross-reference with ZFIN</h3>
<p>This was achieved by mining <a href="./session.html#Zebrafish Information Network (ZFIN)">ZFIN gene expression data</a> for called markers whose expression were detected in the <code>epiphysis</code>. Of course, this is not fool-proof and heavily depends on the accuracy and completion of ZFIN’s annotation. There may be many genes whose expression in the epiphysis were not annotated as such and therefore will be missed (more likely), and <em>vice versa</em> (exception rather than rule).</p>
<p>Follow the link above for the code to pull ZFIN data-of-interest into a dataframe <code>gex_ISH</code> used below :</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>gex <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">url</span>(<span class="st">"https://zfin.org/downloads/wildtype-expression_fish.txt"</span>), <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gex, <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 V1   V2 V3          V4             V5 V6 V7
1 ZDB-GENE-060824-3 a1cf WT ZFA:0000123          liver      
2 ZDB-GENE-060824-3 a1cf WT ZFA:0000123          liver      
3 ZDB-GENE-060824-3 a1cf WT ZFA:0001094 whole organism      
4 ZDB-GENE-090212-1 a2ml WT ZFA:0000112            gut      
5 ZDB-GENE-090212-1 a2ml WT ZFA:0000123          liver      
                       V8               V9                        V10
1                   Adult            Adult  Reverse transcription PCR
2 Larval:Protruding-mouth     Larval:Day 5 mRNA in situ hybridization
3                   Adult            Adult  Reverse transcription PCR
4        Larval:Days 7-13 Larval:Days 7-13 mRNA in situ hybridization
5            Larval:Day 4     Larval:Day 4 mRNA in situ hybridization
          V11               V12 V13 V14                   V15
1 MMO:0000655  ZDB-PUB-060501-4         ZDB-FISH-150901-29105
2 MMO:0000658  ZDB-PUB-060501-4         ZDB-FISH-150901-29105
3 MMO:0000655  ZDB-PUB-060501-4         ZDB-FISH-150901-29105
4 MMO:0000658 ZDB-PUB-081121-13         ZDB-FISH-150901-29105
5 MMO:0000658 ZDB-PUB-081121-13         ZDB-FISH-150901-29105</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add column IDs (based on Column Headers in the Downloads page above)</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gex) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>,<span class="st">"FishName"</span>,<span class="st">"SuperStructureID"</span>,<span class="st">"SuperStructureName"</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"SubStructureID"</span>,<span class="st">"SubStructureName"</span>,<span class="st">"StartStage"</span>,<span class="st">"EndStage"</span>,<span class="st">"Assay"</span>,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"AssayMMOID"</span>,<span class="st">"PublicationID"</span>,<span class="st">"ProbeID"</span>,<span class="st">"AntibodyID"</span>,<span class="st">"FishID"</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>gex_ISH <span class="ot">&lt;-</span> gex[<span class="fu">which</span>(gex<span class="sc">$</span>FishName<span class="sc">==</span><span class="st">'WT'</span> <span class="sc">|</span> gex<span class="sc">$</span>FishName<span class="sc">==</span><span class="st">'AB/TU'</span>), ]</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>gex_ISH <span class="ot">&lt;-</span> gex_ISH[<span class="fu">which</span>(gex_ISH<span class="sc">$</span>Assay<span class="sc">==</span><span class="st">'mRNA in situ hybridization'</span>), ]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>gex_ISH <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(gex_ISH<span class="sc">$</span>GeneSymbol, gex_ISH<span class="sc">$</span>SuperStructureName)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gex_ISH) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GeneSymbol"</span>,<span class="st">"Structure"</span>)</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gex_ISH, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   GeneSymbol            Structure
1        a1cf                liver
2        a2ml                  gut
3        a2ml                liver
4        a2ml                liver
5        a2ml                liver
6        a2ml                 yolk
7        a2ml                 yolk
8        a2ml yolk syncytial layer
9        a2ml yolk syncytial layer
10       a2ml yolk syncytial layer</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Match every marker with every known WISH-detected Structure</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>cl2.de.markers.epiphysis <span class="ot">&lt;-</span> <span class="fu">left_join</span>(cl2.de.markers, gex_ISH, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"markers"</span> <span class="ot">=</span> <span class="st">"GeneSymbol"</span>))</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cl2.de.markers.epiphysis, <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  p_val avg_log2FC pct.1 pct.2 p_val_adj markers                   Structure
1     0   2.755492 0.938   0.1         0    otx1  alar plate midbrain region
2     0   2.755492 0.938   0.1         0    otx1    anterior axial hypoblast
3     0   2.755492 0.938   0.1         0    otx1                        axis
4     0   2.755492 0.938   0.1         0    otx1 basal plate midbrain region
5     0   2.755492 0.938   0.1         0    otx1                diencephalon</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for "epiphysis" only</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>cl2.de.markers.epiphysis <span class="ot">&lt;-</span> <span class="fu">distinct</span>(<span class="fu">filter</span>(cl2.de.markers.epiphysis, Structure<span class="sc">==</span><span class="st">"epiphysis"</span>))</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cl2.de.markers.epiphysis, <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          p_val avg_log2FC pct.1 pct.2     p_val_adj markers Structure
1  0.000000e+00   2.755492 0.938 0.100  0.000000e+00    otx1 epiphysis
2  0.000000e+00   3.250256 0.810 0.044  0.000000e+00   otx2b epiphysis
3  0.000000e+00   2.263520 0.821 0.097  0.000000e+00   otx2a epiphysis
4  0.000000e+00   2.333122 0.783 0.076  0.000000e+00    wnt4 epiphysis
5 4.197046e-284   2.721522 0.959 0.210 1.005025e-279  cldn5a epiphysis</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(cl2.de.markers.epiphysis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 135</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Match every marker with every known WISH-detected Structure</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>cl19.de.markers.epiphysis <span class="ot">&lt;-</span> <span class="fu">left_join</span>(cl19.de.markers, gex_ISH, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"markers"</span> <span class="ot">=</span> <span class="st">"GeneSymbol"</span>))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cl19.de.markers.epiphysis, <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  p_val avg_log2FC pct.1 pct.2 p_val_adj markers         Structure
1     0     5.8017  0.91  0.03         0     bsx             brain
2     0     5.8017  0.91  0.03         0     bsx             brain
3     0     5.8017  0.91  0.03         0     bsx             brain
4     0     5.8017  0.91  0.03         0     bsx caudal tuberculum
5     0     5.8017  0.91  0.03         0     bsx caudal tuberculum</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for "epiphysis" only</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>cl19.de.markers.epiphysis <span class="ot">&lt;-</span> <span class="fu">distinct</span>(<span class="fu">filter</span>(cl19.de.markers.epiphysis, Structure<span class="sc">==</span><span class="st">"epiphysis"</span>))</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cl19.de.markers.epiphysis, <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          p_val avg_log2FC pct.1 pct.2     p_val_adj markers Structure
1  0.000000e+00   5.801700 0.910 0.030  0.000000e+00     bsx epiphysis
2 3.397695e-248   6.210768 0.742 0.027 8.136121e-244   nxph1 epiphysis
3 2.688572e-217   6.833524 0.337 0.003 6.438054e-213  rcvrn2 epiphysis
4 9.997499e-179   3.981461 0.865 0.062 2.394001e-174  esrrga epiphysis
5 4.366307e-176   5.705932 0.404 0.007 1.045556e-171  sall3a epiphysis</code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(cl19.de.markers.epiphysis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 302</code></pre>
</div>
</div>
<p>This step identifies a significantly more manageable ~100 and 300 candidate genes per cluster for a MERFISH experiment.</p>
<ul>
<li><p><a href="https://github.com/vchongmorrison/zfBrainAsymmetry/blob/master/annotations/cl2_de_markers_epiphysis.txt">Link</a> to Cluster <code>2</code> candidate gene list</p></li>
<li><p><a href="https://github.com/vchongmorrison/zfBrainAsymmetry/blob/master/annotations/cl19_de_markers_epiphysis.txt">Link</a> to Cluster <code>19</code> candidate gene list</p></li>
</ul>
<p>At this point, the gene lists were manually shortlisted for inclusion into the final MERSCOPE 140 Gene Panel, with decreasing priority below.</p>
<ol type="1">
<li><p>Genes from the literature/previous work</p>
<ul>
<li><p>Clanton <em>et al</em>. 2013</p></li>
<li><p>Khuansuwan <em>et al</em>. 2016</p></li>
<li><p>Schredelseker &amp; Driever, 2018</p></li>
</ul></li>
<li><p>Genes related to GABA synthesis and signalling</p>
<ul>
<li>(one-liner explanation from S Wilson? TBC)</li>
</ul></li>
<li><p>Selected from Cluster <code>2</code> and <code>19</code> candidate gene lists, that generally have some previous links to neurogenesis</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save gene lists to file</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(cl2.de.markers.epiphysis<span class="sc">$</span>markers), <span class="at">file=</span><span class="st">"./annotations/cl2_de_markers_epiphysis.txt"</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(cl19.de.markers.epiphysis<span class="sc">$</span>markers), <span class="at">file=</span><span class="st">"./annotations/cl19_de_markers_epiphysis.txt"</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="gene-panel-for-merscope---final" class="level3" data-number="2.3">
<h3 data-number="2.3" class="anchored" data-anchor-id="gene-panel-for-merscope---final"><span class="header-section-number">2.3</span> 140 Gene Panel for MERSCOPE - final</h3>
<ul>
<li><a href="https://github.com/vchongmorrison/zfBrainAsymmetry/blob/master/annotations/codebook_0_AM0765_zebrafishChongMorrisonAsymmetry.csv">Link</a> to panel codebook</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>merscope <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'./annotations/codebook_0_AM0765_zebrafishChongMorrisonAsymmetry.csv'</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(merscope, <span class="dv">10</span>) <span class="co"># print first 10 rows only</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                        id barcodeType V0001T8B1 V0002T8B1 V0003T8B1 V0004T8B1
aanat2  ENSDART00000018205     merfish         0         0         1         0
alcama  ENSDART00000129400     merfish         0         0         0         0
aldocb  ENSDART00000026766     merfish         0         1         0         1
arr3a   ENSDART00000078996     merfish         0         0         0         1
arr3b   ENSDART00000158077     merfish         0         0         0         0
ascl1a  ENSDART00000056005     merfish         0         0         0         0
asip2b  ENSDART00000185699     merfish         0         0         0         0
atp1a3a ENSDART00000104950     merfish         0         0         0         1
atp1b1b ENSDART00000139387     merfish         0         0         0         0
atp1b2a ENSDART00000160044     merfish         0         1         0         0
        V0005T8B1 V0006T8B1 V0007T8B1 V0008T8B1 V0009T8B1 V0010T8B1 V0011T8B1
aanat2          0         0         0         0         0         0         1
alcama          0         0         0         0         0         0         0
aldocb          0         0         0         0         1         0         0
arr3a           0         0         0         1         0         0         0
arr3b           1         1         1         0         1         0         0
ascl1a          0         0         1         0         1         0         0
asip2b          1         0         1         0         0         0         0
atp1a3a         0         0         0         0         0         1         1
atp1b1b         0         0         0         1         0         1         0
atp1b2a         0         0         0         0         0         0         0
        V0012T8B1 V0013T8B1 V0014T8B1 V0015T8B1 V0016T8B1 V0017T8B1 V0018T8B1
aanat2          0         0         0         1         1         0         0
alcama          0         1         1         1         0         1         0
aldocb          0         0         0         1         0         0         0
arr3a           0         0         1         0         0         1         0
arr3b           0         0         0         0         0         0         0
ascl1a          0         0         0         1         0         1         0
asip2b          0         1         0         1         0         0         0
atp1a3a         0         0         0         0         1         0         0
atp1b1b         0         1         0         0         0         1         0
atp1b2a         0         1         0         1         0         0         1</code></pre>
</div>
</div>
<section id="composition-of-the-final-panel" class="level4" data-number="2.3.1">
<h4 data-number="2.3.1" class="anchored" data-anchor-id="composition-of-the-final-panel"><span class="header-section-number">2.3.1</span> <strong>Composition of the final panel</strong></h4>
<p><a href="https://github.com/hms-dbmi/UpSetR">UpSet</a> plots are an excellent alternative to Venn diagrams for visualising intersection between the three gene lists above.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="do">##### Explore intersection of markers between clusters #####</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UpSetR)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>cl2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"./annotations/cl2_de_markers_epiphysis.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>cl2<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="st">"Cluster2"</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cl2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"markers"</span>,<span class="st">"cluster"</span>)</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>cl19 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"./annotations/cl19_de_markers_epiphysis.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>cl19<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="st">"Cluster19"</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cl19) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"markers"</span>,<span class="st">"cluster"</span>)</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>merscope <span class="ot">&lt;-</span> merscope[<span class="dv">1</span><span class="sc">:</span><span class="dv">140</span>,] <span class="co"># 140 gene targets, remove 40 Blanks</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>merscope<span class="sc">$</span>markers <span class="ot">&lt;-</span> <span class="fu">rownames</span>(merscope)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>merscope<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="st">"merscope"</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>merscope <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(merscope<span class="sc">$</span>markers, merscope<span class="sc">$</span>cluster)</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(merscope) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"markers"</span>,<span class="st">"cluster"</span>)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>all.markers <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cl2, cl19, merscope)</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Format data to serve as input for UpSetR.</span></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>input <span class="ot">&lt;-</span> all.markers <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">truval=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">spread</span>(cluster, truval, <span class="at">fill=</span><span class="cn">FALSE</span>)</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>input <span class="ot">&lt;-</span> input <span class="sc">%&gt;%</span></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>, <span class="sc">~</span> <span class="fu">as.integer</span>(<span class="fu">as.character</span>(<span class="fu">factor</span>(., <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>))))))</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a><span class="co"># UpSetR plot - Conway et al. 2017</span></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a><span class="fu">upset</span>(input, <span class="at">empty.intersections =</span> <span class="st">"on"</span>, <span class="at">order.by =</span> <span class="st">"freq"</span>, <span class="at">mainbar.y.label =</span> <span class="st">"Epiphysis marker intersections"</span>, <span class="at">sets.x.label =</span> <span class="st">"Epiphysis markers per set"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part2_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Of the 140 genes in the MERSCOPE panel;</p>
<ul>
<li><p>The large majority, 103, are specific to Cluster <code>19</code> candidate genes</p></li>
<li><p>9 are shared between Cluster <code>2</code> and <code>19</code></p></li>
<li><p>4 are specific to Cluster <code>2</code> candidate genes</p></li>
<li><p>24 are from neither i.e.&nbsp;from the literature and GABA genes</p></li>
</ul>
</section>
<section id="expression-pattern-in-ss_e107" class="level4" data-number="2.3.2">
<h4 data-number="2.3.2" class="anchored" data-anchor-id="expression-pattern-in-ss_e107"><span class="header-section-number">2.3.2</span> <strong>Expression pattern in <code>ss_e107</code></strong></h4>
<p>To explore the expression of these genes in the <code>ss_e107</code> scRNA-seq dataset, let’s first check that all the panel genes are present in the <code>SeuratObject</code>. Otherwise, plotting will throw errors.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MERSCOPE panel genes</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>merscope <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"./annotations/codebook_0_AM0765_zebrafishChongMorrisonAsymmetry.csv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>merscope <span class="ot">&lt;-</span> merscope[<span class="dv">1</span><span class="sc">:</span><span class="dv">140</span>,]</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 genes filtered out in ss_e107 dataset, remove these otherwise plotting ss_e107 will throw error</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rownames</span>(ss_e107))</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(genes) <span class="ot">&lt;-</span> <span class="st">"gene"</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>merscope_check <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(genes, merscope, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"name"</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(barcodeType<span class="sc">==</span><span class="st">"merfish"</span>) <span class="co"># 137 genes</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="fu">anti_join</span>(merscope, merscope_check, <span class="at">by=</span><span class="fu">join_by</span>(name <span class="sc">==</span> gene)) <span class="co"># to get names of these genes (gfi1ab, opn1sw1, rho)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     name                 id barcodeType V0001T8B1 V0002T8B1 V0003T8B1
1  gfi1ab ENSDART00000065139     merfish         0         0         0
2 opn1sw1 ENSDART00000067160     merfish         1         0         0
3     rho ENSDART00000027000     merfish         1         0         0
  V0004T8B1 V0005T8B1 V0006T8B1 V0007T8B1 V0008T8B1 V0009T8B1 V0010T8B1
1         1         1         0         0         0         0         0
2         0         0         0         1         0         0         1
3         0         0         0         0         0         0         0
  V0011T8B1 V0012T8B1 V0013T8B1 V0014T8B1 V0015T8B1 V0016T8B1 V0017T8B1
1         0         0         1         1         0         0         0
2         0         0         0         0         0         0         1
3         0         1         0         1         0         0         1
  V0018T8B1
1         0
2         0
3         0</code></pre>
</div>
</div>
<p>Plot a heatmap of the 140 (137) panel genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoHeatmap</span>(ss_e107, merscope_check<span class="sc">$</span>gene,</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">annot.by =</span> <span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">scaled.to.max =</span> <span class="cn">TRUE</span>,</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, <span class="at">complex=</span><span class="cn">TRUE</span>, <span class="at">fontsize_row =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part2_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This heatmap illustrates the on-going challenge - that the epiphysis shares many of its expressed genes with other neuronal and/or cranial cell types. Despite ‘marker identification’ followed by filtering for those with pre-validated <em>in vivo</em> expression in the epiphysis, these genes as a whole <strong><em>do not specifically</em></strong> mark out the (putative) epiphysis clusters <code>2</code> and <code>19</code>. Therefore, from a bioinformatics perspective, using a <strong><em>combination of genes</em></strong> is vital to ‘extract’ these cells - as demonstrated in <a href="./Aim1_part1.html#Identify epiphysis clusters">Aim 1 Part 1: Identify epiphysis clusters</a>.</p>
</section>
</section>
<section id="additional-insights" class="level3" data-number="2.4">
<h3 data-number="2.4" class="anchored" data-anchor-id="additional-insights"><span class="header-section-number">2.4</span> Additional insights</h3>
<p>However, in the painstaking iterative process of going through the gene lists above and cross-checking with ZFIN, two groups of genes were noted to display specific expression in the epiphysis and retina. The ones highlighted in <strong>bold</strong> were called in Clusters <code>2</code> and <code>19</code> candidate gene lists.</p>
<ul>
<li><p>Recoverins - <code>rcvrna</code>, <code>rcvrnb</code>, <strong><code>rcvrn2</code></strong>, <code>rcvrn3</code></p></li>
<li><p>Phosphodiesterases - <strong><code>pde6gb</code></strong>,<strong><code>pde6c</code></strong>,<strong><code>pde6d</code></strong> … (too many to list).</p></li>
</ul>
<p>The <a href="https://zfin.org/ZDB-IMAGE-151008-27#image">recoverins</a> and <a href="https://zfin.org/ZDB-FIG-050630-4919#summary"><em>pde6gb</em></a> in particular had striking epiphysis- plus/minus retina-specific expression patterns in the embryo. They reflect the photoreceptive characteristics of these organs. Let’s look at these genes together with the three markers previously used to identify epiphysis clusters - <em>bsx</em>, <em>noto</em>, <em>otx5</em>. <em>Sox1a</em> is included as the parapineal marker.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>epiphysis_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"bsx"</span>, <span class="st">"noto"</span>, <span class="st">"otx5"</span>, <span class="st">"sox1a"</span>,</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"rcvrna"</span>, <span class="st">"rcvrnb"</span>, <span class="st">"rcvrn2"</span>, <span class="st">"rcvrn3"</span>,</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"pde6gb"</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDotPlot</span>(ss_e107, epiphysis_genes, <span class="at">group.by=</span><span class="st">"SCT_snn_res.1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part2_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Cluster <code>19</code> demonstrated robust expression of several of these ‘ZFIN-validated’ genes. Remarkably, the remaining clusters, including Cluster <code>2</code>, showed co-expression of these genes as well to a certain extent…are these ‘contaminating’ retinal photoreceptor precursors? However, the parapineal marker <em>sox1a</em> is still only detected in Clusters <code>2</code> and <code>19</code>.</p>
<p>With these lessons in mind, we proceed with the MERSCOPE experiment to gather additional evidence and wrap-up Aim 1.</p>
</section>
</section>
<section id="pre-processing-of-merscope-data" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="pre-processing-of-merscope-data"><span class="header-section-number">3</span> Pre-processing of MERSCOPE data</h2>
<p><a href="https://vizgen.com/vizualizer-software/">MERSCOPE Vizualizer</a> is a free software provided by Vizgen as an out-of-box solution to explore results from MERSCOPE runs. The following images are exported using the following settings: <code>Transcripts Z-plane = All</code>; <code>Z-plane = 2</code> ; <code>Images = DAPI</code> channel only. The MERFISH signal coverage clearly encompasses the epiphysis region, which is promising. However, these images also have the <code>cell</code> setting ON…and it is clear that the default cell segmentation algorithm in the run is incompatible with the samples.</p>
<div id="fig-e1" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/embryo1_ant_all-genes_DAPI.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Embryo 1 - anterior slice</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/embryo1_pos_all-genes_DAPI.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Embryo 1 - posterior slice</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Embryo 1 with all detected transcripts and failed cell segmentation</figcaption><p></p>
</figure>
</div>
<div id="fig-e2" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/embryo2_ant_all-genes_DAPI.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Embryo 2 - anterior slice</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/embryo2_pos_all-genes_DAPI.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Embryo 2 - posterior slice</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;2: Embryo 2 with all detected transcripts and failed cell segmentation</figcaption><p></p>
</figure>
</div>
<section id="cell-segmentation" class="level3" data-number="3.1">
<h3 data-number="3.1" class="anchored" data-anchor-id="cell-segmentation"><span class="header-section-number">3.1</span> Cell segmentation</h3>
<p>Through a lot of trial and error, I worked out that <a href="https://github.com/MouseLand/cellpose">Cellpose</a>‘s <code>cyto3</code> model released with v3 (Feb 2024) had the best performance in segmenting these images. In addition, Vizgen’s <a href="https://vizgen.com/vpt/">VPT</a> tool with the <a href="https://github.com/Vizgen/vpt-plugin-cellpose2">Cellpose2 plugin</a> enables performing custom 2-task segmentation runs, where the results of two (up to six) segmentation settings can be harmonised for a ’consensus’-like result (and perhaps improve sensitivity). The final pipeline is implemented as an <code>apptainer</code> container that runs VPT-Cellpose2 to perform 2-task segmentation using the <code>cyto3</code> model.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># apptainer.def file</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-reseg" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/embryo1_pos_all-genes_DAPI_reseg.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Embryo 1 - posterior slice</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/embryo2_pos_all-genes_DAPI_reseg.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Embryo 2 - posterior slice</figcaption>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;3: Embryo 1 and 2 with all detected transcripts and successful re-segmentation</figcaption><p></p>
</figure>
</div>
</section>
</section>
<section id="clustering-of-merscope-data" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="clustering-of-merscope-data"><span class="header-section-number">4</span> Clustering of MERSCOPE data</h2>
<p>in progress</p>
<hr>
</section>
<section id="summary-notes-for-aim-1---part-2" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="summary-notes-for-aim-1---part-2"><span class="header-section-number">5</span> Summary notes for Aim 1 - Part 2</h2>
<ol type="1">
<li><p>in progress</p></li>
<li><p>in progress</p></li>
</ol>
<hr>
</section>
<section id="references" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="references"><span class="header-section-number">6</span> References</h2>
<section id="packages" class="level3" data-number="6.1">
<h3 data-number="6.1" class="anchored" data-anchor-id="packages"><span class="header-section-number">6.1</span> Packages</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>pkgs <span class="ot">&lt;-</span> <span class="fu">cite_packages</span>(<span class="at">output =</span> <span class="st">"table"</span>, <span class="at">out.dir =</span> <span class="st">"."</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(pkgs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Package</th>
<th style="text-align: left;">Version</th>
<th style="text-align: left;">Citation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">base</td>
<td style="text-align: left;">4.4.2</td>
<td style="text-align: left;"><span class="citation" data-cites="base">R Core Team (<a href="#ref-base" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">BiocParallel</td>
<td style="text-align: left;">1.38.0</td>
<td style="text-align: left;"><span class="citation" data-cites="BiocPar">(<a href="#ref-BiocPar" role="doc-biblioref"><strong>BiocPar?</strong></a>)</span>….</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dittoSeq</td>
<td style="text-align: left;">1.16.0</td>
<td style="text-align: left;"><span class="citation" data-cites="dittoSeq">Bunis et al. (<a href="#ref-dittoSeq" role="doc-biblioref">2020</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">future</td>
<td style="text-align: left;">1.34.0</td>
<td style="text-align: left;"><span class="citation" data-cites="RJ-2021-048">Bengtsson (<a href="#ref-RJ-2021-048" role="doc-biblioref">2021</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">knitr</td>
<td style="text-align: left;">1.49</td>
<td style="text-align: left;"><span class="citation" data-cites="knitr20">(<a href="#ref-knitr20" role="doc-biblioref"><strong>knitr20?</strong></a>)</span>….</td>
</tr>
<tr class="even">
<td style="text-align: left;">Matrix</td>
<td style="text-align: left;">1.7.0</td>
<td style="text-align: left;"><span class="citation" data-cites="Matrix">Bates, Maechler, and Jagan (<a href="#ref-Matrix" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">RCurl</td>
<td style="text-align: left;">1.98.1.16</td>
<td style="text-align: left;"><span class="citation" data-cites="RCurl">Temple Lang (<a href="#ref-RCurl" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">reticulate</td>
<td style="text-align: left;">1.40.0</td>
<td style="text-align: left;"><span class="citation" data-cites="reticulate">Ushey, Allaire, and Tang (<a href="#ref-reticulate" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">rmarkdown</td>
<td style="text-align: left;">2.29</td>
<td style="text-align: left;"><span class="citation" data-cites="rmarkdo">(<a href="#ref-rmarkdo" role="doc-biblioref"><strong>rmarkdo?</strong></a>)</span>….</td>
</tr>
<tr class="even">
<td style="text-align: left;">scDblFinder</td>
<td style="text-align: left;">1.18.0</td>
<td style="text-align: left;"><span class="citation" data-cites="scDblFinder">Germain et al. (<a href="#ref-scDblFinder" role="doc-biblioref">2022</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Seurat</td>
<td style="text-align: left;">5.1.0</td>
<td style="text-align: left;"><span class="citation" data-cites="Seurat2">(<a href="#ref-Seurat2" role="doc-biblioref"><strong>Seurat2?</strong></a>)</span>….</td>
</tr>
<tr class="even">
<td style="text-align: left;">SingleCellExperiment</td>
<td style="text-align: left;">1.26.0</td>
<td style="text-align: left;"><span class="citation" data-cites="SingleC">(<a href="#ref-SingleC" role="doc-biblioref"><strong>SingleC?</strong></a>)</span>….</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tidyverse</td>
<td style="text-align: left;">2.0.0</td>
<td style="text-align: left;"><span class="citation" data-cites="tidyverse">Wickham et al. (<a href="#ref-tidyverse" role="doc-biblioref">2019</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">UpSetR</td>
<td style="text-align: left;">1.4.0</td>
<td style="text-align: left;"><span class="citation" data-cites="UpSetR">Gehlenborg (<a href="#ref-UpSetR" role="doc-biblioref">2019</a>)</span></td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="previous-work" class="level3" data-number="6.2">
<h3 data-number="6.2" class="anchored" data-anchor-id="previous-work"><span class="header-section-number">6.2</span> Previous work</h3>
<ul>
<li>I. Lekk, V. Duboc, A. Faro, S. Nicolaou, P. Blader and S.W. Wilson. <em>Sox1a mediates the ability of the parapineal to impart habenular left-right asymmetry</em>. eLife (2019), e47376, 8, <a href="https://doi.org/10.7554/eLife.47376" class="uri">https://doi.org/10.7554/eLife.47376</a></li>
<li>S. Khuansuwan, J. A. Clanton, B. J. Dean, J. G. Patton, J. T. Gamse; A transcription factor network controls cell migration and fate decisions in the developing zebrafish pineal complex. <em>Development</em> 15 July 2016; 143 (14): 2641–2650, <a href="https://doi.org/10.1242/dev.131680" class="uri">https://doi.org/10.1242/dev.131680</a></li>
<li>J. A. Clanton, K. D. Hope, J. T. Gamse; Fgf signaling governs cell fate in the zebrafish pineal complex. <em>Development</em> 15 January 2013; 140 (2): 323–332, <a href="https://doi.org/10.1242/dev.083709" class="uri">https://doi.org/10.1242/dev.083709</a></li>
<li>Theresa Schredelseker, Wolfgang Driever; Bsx controls pineal complex development. <em>Development</em> 1 July 2018; 145 (13): dev163477, <a href="https://doi.org/10.1242/dev.163477" class="uri">https://doi.org/10.1242/dev.163477</a></li>
</ul>
</section>
</section>
<section id="session-info" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="session-info"><span class="header-section-number">7</span> Session Info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.2 (2024-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/London
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] UpSetR_1.4.0                grateful_0.2.10            
 [3] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
 [5] Biobase_2.64.0              GenomicRanges_1.56.2       
 [7] GenomeInfoDb_1.40.1         IRanges_2.38.1             
 [9] S4Vectors_0.42.1            BiocGenerics_0.50.0        
[11] MatrixGenerics_1.16.0       matrixStats_1.4.1          
[13] dittoSeq_1.16.0             lubridate_1.9.3            
[15] forcats_1.0.0               stringr_1.5.1              
[17] dplyr_1.1.4                 purrr_1.0.2                
[19] readr_2.1.5                 tidyr_1.3.1                
[21] tibble_3.2.1                ggplot2_3.5.1              
[23] tidyverse_2.0.0             Seurat_5.1.0               
[25] SeuratObject_5.0.2          sp_2.1-4                   
[27] reticulate_1.40.0           BiocParallel_1.38.0        
[29] future_1.34.0              

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22        splines_4.4.2           later_1.3.2            
  [4] polyclip_1.10-7         fastDummies_1.7.4       lifecycle_1.0.4        
  [7] doParallel_1.0.17       globals_0.16.3          lattice_0.22-6         
 [10] MASS_7.3-61             magrittr_2.0.3          limma_3.60.6           
 [13] plotly_4.10.4           rmarkdown_2.29          yaml_2.3.10            
 [16] httpuv_1.6.15           sctransform_0.4.1       spam_2.11-0            
 [19] spatstat.sparse_3.1-0   cowplot_1.1.3           pbapply_1.7-2          
 [22] RColorBrewer_1.1-3      abind_1.4-8             zlibbioc_1.50.0        
 [25] Rtsne_0.17              presto_1.0.0            rappdirs_0.3.3         
 [28] circlize_0.4.16         GenomeInfoDbData_1.2.12 ggrepel_0.9.6          
 [31] irlba_2.3.5.1           listenv_0.9.1           spatstat.utils_3.1-1   
 [34] pheatmap_1.0.12         goftest_1.2-3           RSpectra_0.16-2        
 [37] spatstat.random_3.3-2   fitdistrplus_1.2-1      parallelly_1.39.0      
 [40] leiden_0.4.3.1          codetools_0.2-20        DelayedArray_0.30.1    
 [43] shape_1.4.6.1           tidyselect_1.2.1        UCSC.utils_1.0.0       
 [46] farver_2.1.2            spatstat.explore_3.3-3  jsonlite_1.8.9         
 [49] GetoptLong_1.0.5        progressr_0.15.0        ggridges_0.5.6         
 [52] survival_3.7-0          iterators_1.0.14        foreach_1.5.2          
 [55] tools_4.4.2             ica_1.0-3               Rcpp_1.0.13-1          
 [58] glue_1.8.0              gridExtra_2.3           SparseArray_1.4.8      
 [61] xfun_0.49               withr_3.0.2             fastmap_1.2.0          
 [64] fansi_1.0.6             digest_0.6.37           timechange_0.3.0       
 [67] R6_2.5.1                mime_0.12               colorspace_2.1-1       
 [70] Cairo_1.6-2             scattermore_1.2         tensor_1.5             
 [73] spatstat.data_3.1-2     utf8_1.2.4              generics_0.1.3         
 [76] renv_1.0.11             data.table_1.16.2       httr_1.4.7             
 [79] htmlwidgets_1.6.4       S4Arrays_1.4.1          uwot_0.2.2             
 [82] pkgconfig_2.0.3         gtable_0.3.6            ComplexHeatmap_2.20.0  
 [85] lmtest_0.9-40           XVector_0.44.0          htmltools_0.5.8.1      
 [88] dotCall64_1.2           clue_0.3-66             scales_1.3.0           
 [91] png_0.1-8               spatstat.univar_3.1-1   knitr_1.49             
 [94] rstudioapi_0.17.1       rjson_0.2.23            tzdb_0.4.0             
 [97] reshape2_1.4.4          nlme_3.1-165            zoo_1.8-12             
[100] GlobalOptions_0.1.2     KernSmooth_2.23-24      parallel_4.4.2         
[103] miniUI_0.1.1.1          pillar_1.9.0            grid_4.4.2             
[106] vctrs_0.6.5             RANN_2.6.2              promises_1.3.0         
[109] xtable_1.8-4            cluster_2.1.6           evaluate_1.0.1         
[112] cli_3.6.3               compiler_4.4.2          rlang_1.1.4            
[115] crayon_1.5.3            future.apply_1.11.3     labeling_0.4.3         
[118] plyr_1.8.9              stringi_1.8.4           viridisLite_0.4.2      
[121] deldir_2.0-4            munsell_0.5.1           lazyeval_0.2.2         
[124] spatstat.geom_3.3-3     Matrix_1.7-0            RcppHNSW_0.6.0         
[127] hms_1.1.3               patchwork_1.3.0         statmod_1.5.0          
[130] shiny_1.9.1             ROCR_1.0-11             igraph_2.1.1           </code></pre>
</div>
</div>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-Matrix" class="csl-entry" role="listitem">
Bates, Douglas, Martin Maechler, and Mikael Jagan. 2024. <em><span>Matrix</span>: Sparse and Dense Matrix Classes and Methods</em>. <a href="https://CRAN.R-project.org/package=Matrix">https://CRAN.R-project.org/package=Matrix</a>.
</div>
<div id="ref-RJ-2021-048" class="csl-entry" role="listitem">
Bengtsson, Henrik. 2021. <span>“A Unifying Framework for Parallel and Distributed Processing in r Using Futures.”</span> <em>The R Journal</em> 13 (2): 208–27. <a href="https://doi.org/10.32614/RJ-2021-048">https://doi.org/10.32614/RJ-2021-048</a>.
</div>
<div id="ref-dittoSeq" class="csl-entry" role="listitem">
Bunis, Daniel G., Jared Andrews, Gabriela K. Fragiadakis, Trevor D. Burt, and Marina Sirota. 2020. <span>“<span class="nocase">dittoSeq</span>: Universal User-Friendly Single-Cell and Bulk RNA Sequencing Visualization Toolkit.”</span> <em>Bioinformatics</em>, no. 22-23 (December): 5535–36. <a href="https://doi.org/10.1093/bioinformatics/btaa1011">https://doi.org/10.1093/bioinformatics/btaa1011</a>.
</div>
<div id="ref-UpSetR" class="csl-entry" role="listitem">
Gehlenborg, Nils. 2019. <em><span>UpSetR</span>: A More Scalable Alternative to Venn and Euler Diagrams for Visualizing Intersecting Sets</em>. <a href="https://CRAN.R-project.org/package=UpSetR">https://CRAN.R-project.org/package=UpSetR</a>.
</div>
<div id="ref-scDblFinder" class="csl-entry" role="listitem">
Germain, Pierre-Luc, Aaron Lun, Carlos Garcia Meixide, Will Macnair, and Mark D. Robinson. 2022. <span>“Doublet Identification in Single-Cell Sequencing Data Using scDblFinder.”</span> <em>F1000research</em>. <a href="https://doi.org/10.12688/f1000research.73600.2">https://doi.org/10.12688/f1000research.73600.2</a>.
</div>
<div id="ref-base" class="csl-entry" role="listitem">
R Core Team. 2024. <em><span>R</span>: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-RCurl" class="csl-entry" role="listitem">
Temple Lang, Duncan. 2024. <em><span>RCurl</span>: General Network (HTTP/FTP/...) Client Interface for r</em>. <a href="https://CRAN.R-project.org/package=RCurl">https://CRAN.R-project.org/package=RCurl</a>.
</div>
<div id="ref-reticulate" class="csl-entry" role="listitem">
Ushey, Kevin, JJ Allaire, and Yuan Tang. 2024. <em><span class="nocase">reticulate</span>: Interface to <span>“<span>Python</span>”</span></em>. <a href="https://CRAN.R-project.org/package=reticulate">https://CRAN.R-project.org/package=reticulate</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span class="nocase">tidyverse</span>.”</span> <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://utteranc.es/client.js" repo="zfBrainAsymmetry/comments" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">Copyright 2024, Vanessa Chong-Morrison</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>