<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>zfBrainAsymmetry - The parapineal ‘cell type’ - Part 1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">zfBrainAsymmetry</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./session.html" rel="" target="">
 <span class="menu-text">Session</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-aims" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Aims</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-aims">    
        <li>
    <a class="dropdown-item" href="./Aim1_part1.html" rel="" target="">
 <span class="dropdown-text">The parapineal ‘cell type’ - Part 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Aim2.html" rel="" target="">
 <span class="dropdown-text">The sox1a ‘phenotype’</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#identification-of-epiphysis-cells" id="toc-identification-of-epiphysis-cells" class="nav-link active" data-scroll-target="#identification-of-epiphysis-cells">Identification of epiphysis cells</a>
  <ul class="collapse">
  <li><a href="#premise" id="toc-premise" class="nav-link" data-scroll-target="#premise"><strong>Premise</strong></a></li>
  <li><a href="#experimental-design" id="toc-experimental-design" class="nav-link" data-scroll-target="#experimental-design">Experimental design</a></li>
  <li><a href="#initialise-r-environment" id="toc-initialise-r-environment" class="nav-link" data-scroll-target="#initialise-r-environment">Initialise R environment</a></li>
  <li><a href="#pre-processing-of-reads" id="toc-pre-processing-of-reads" class="nav-link" data-scroll-target="#pre-processing-of-reads">Pre-processing of reads</a></li>
  </ul></li>
  <li><a href="#scrna-seq-analysis-in-seurat" id="toc-scrna-seq-analysis-in-seurat" class="nav-link" data-scroll-target="#scrna-seq-analysis-in-seurat">scRNA-seq analysis in <code>Seurat</code></a>
  <ul class="collapse">
  <li><a href="#load-cell-x-gene-matrices" id="toc-load-cell-x-gene-matrices" class="nav-link" data-scroll-target="#load-cell-x-gene-matrices">Load cell x gene matrices</a></li>
  <li><a href="#filter-genes" id="toc-filter-genes" class="nav-link" data-scroll-target="#filter-genes">Filter genes</a></li>
  <li><a href="#compute-additional-metrics---percent.mito-ribo-complexity" id="toc-compute-additional-metrics---percent.mito-ribo-complexity" class="nav-link" data-scroll-target="#compute-additional-metrics---percent.mito-ribo-complexity">Compute additional metrics - percent.mito, ribo, complexity</a></li>
  <li><a href="#compute-cell-cycle-scores" id="toc-compute-cell-cycle-scores" class="nav-link" data-scroll-target="#compute-cell-cycle-scores">Compute cell cycle scores</a></li>
  <li><a href="#cell-filtering" id="toc-cell-filtering" class="nav-link" data-scroll-target="#cell-filtering">Cell filtering?</a></li>
  </ul></li>
  <li><a href="#dimensional-reductionclustering" id="toc-dimensional-reductionclustering" class="nav-link" data-scroll-target="#dimensional-reductionclustering">Dimensional reduction/clustering</a>
  <ul class="collapse">
  <li><a href="#sctransform-normalise-find-variable-genes-regress-unwanted-variation" id="toc-sctransform-normalise-find-variable-genes-regress-unwanted-variation" class="nav-link" data-scroll-target="#sctransform-normalise-find-variable-genes-regress-unwanted-variation">SCTransform (‘normalise, find variable genes, regress unwanted variation’)</a></li>
  <li><a href="#unsupervised-clustering" id="toc-unsupervised-clustering" class="nav-link" data-scroll-target="#unsupervised-clustering">‘Unsupervised’ Clustering</a></li>
  </ul></li>
  <li><a href="#visualise-and-explore-clustering-results-with-dittoseq" id="toc-visualise-and-explore-clustering-results-with-dittoseq" class="nav-link" data-scroll-target="#visualise-and-explore-clustering-results-with-dittoseq">Visualise and explore clustering results with <code>dittoSeq</code></a>
  <ul class="collapse">
  <li><a href="#umap-reduction" id="toc-umap-reduction" class="nav-link" data-scroll-target="#umap-reduction">UMAP reduction</a></li>
  </ul></li>
  <li><a href="#identify-epiphysis-clusters" id="toc-identify-epiphysis-clusters" class="nav-link" data-scroll-target="#identify-epiphysis-clusters">Identify epiphysis clusters</a></li>
  <li><a href="#sanity-checks-on-clustering" id="toc-sanity-checks-on-clustering" class="nav-link" data-scroll-target="#sanity-checks-on-clustering">Sanity checks on clustering</a>
  <ul class="collapse">
  <li><a href="#check-distribution-of-multiplets" id="toc-check-distribution-of-multiplets" class="nav-link" data-scroll-target="#check-distribution-of-multiplets">Check distribution of multiplets</a></li>
  <li><a href="#what-about-other-metrics" id="toc-what-about-other-metrics" class="nav-link" data-scroll-target="#what-about-other-metrics">What about other metrics?</a></li>
  </ul></li>
  <li><a href="#summary-notes-for-aim-1---part-1" id="toc-summary-notes-for-aim-1---part-1" class="nav-link" data-scroll-target="#summary-notes-for-aim-1---part-1">Summary notes for Aim 1 - Part 1</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a>
  <ul class="collapse">
  <li><a href="#packages" id="toc-packages" class="nav-link" data-scroll-target="#packages">Packages</a></li>
  <li><a href="#previous-work" id="toc-previous-work" class="nav-link" data-scroll-target="#previous-work">Previous work</a></li>
  </ul></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session Info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">The parapineal ‘cell type’ - Part 1</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="identification-of-epiphysis-cells" class="level2">
<h2 class="anchored" data-anchor-id="identification-of-epiphysis-cells">Identification of epiphysis cells</h2>
<section id="premise" class="level3">
<h3 class="anchored" data-anchor-id="premise"><strong>Premise</strong></h3>
<p>Using single cell RNA-seq (scRNA-seq), can pineal and parapineal cells be identified from a mixed population regardless of <em>sox1a</em> genotype?</p>
</section>
<section id="experimental-design" class="level3">
<h3 class="anchored" data-anchor-id="experimental-design">Experimental design</h3>
<p>Approximately 230 heads from ~30-32 hpf larvae were collected. Larvae were of mixed <em>sox1a</em> genotype (hets, homs and wild types) obtained by incrossing <em>sox1a</em> hets.</p>
<p>Heads were dissociated into single cell suspension for FACS-based enrichment of GFP++/mCherry- cells. GFP labels the epiphysis and other ‘contaminant’ cell types, while mCherry labels ‘contaminant’ cell types only. The mCherry transgene was pre-determined in prior imaging experiments to not label the parapineal organ.</p>
<p>Approximately 10k sorted events were recorded and collected for scRNA-seq using 10X Genomics 3’ v3.1 GEX chemistry. The library was sequenced to at least 500 million reads, thus yielding a depth of ~100k reads per cell.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./figures/foxd3flh-facs.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Schematic of experiment design.</figcaption>
</figure>
</div>
</section>
<section id="initialise-r-environment" class="level3">
<h3 class="anchored" data-anchor-id="initialise-r-environment">Initialise R environment</h3>
<p><a href="./session.html">Session</a></p>
</section>
<section id="pre-processing-of-reads" class="level3">
<h3 class="anchored" data-anchor-id="pre-processing-of-reads">Pre-processing of reads</h3>
<p>To empirically determine the robustness of called clusters downstream, reads were pre-processed using two different mapper/gene annotation combinations:</p>
<ul>
<li><p>10X Genomics CellRanger v7.1.0 + Lawson Lab annotation (Lawson et al., 2020)</p>
<ul>
<li><p>This pipeline maps unique reads only:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preprocessing - Cell Ranger 7.1.0 (unique reads only)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> count <span class="at">--id</span><span class="op">=</span>foxd3flhGFP <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>--transcriptome=/home/ucbtvmr/Scratch/cellranger/mkref/GRCz11_lawson/GRCz11_v4.3.2_GFP_mCherry <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>--fastqs=/home/ucbtvmr/Scratch/4_scRNAseq/fastq <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>--sample=VCM2-AK30645-AK6234 <span class="dt">\</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>--no-bam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ul></li>
<li><p>STARsolo v2.7.10b + Ensembl release 107 annotation</p>
<ul>
<li><p>This pipeline includes multi-mapping reads assigned using EM algorithm:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preprocessing - STARsolo v2.7.10b (with EM multi-mapped reads)</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">STAR</span> <span class="at">--genomeDir</span> /home/ucbtvmr/ensembl_releases/ensembl107/Danio_rerio.genome.GFP.mCherry/star_2.7.9a <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>--readFilesIn /home/ucbtvmr/Scratch/4_scRNAseq/fastq/VCM2-AK30645-AK6234<span class="pp">*</span>in progress<span class="pp">**</span>_S1_L002_R2_001.fastq.gz <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>/home/ucbtvmr/Scratch/4_scRNAseq/fastq/VCM2-AK30645-AK6234_S1_L002_R1_001.fastq.gz <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>--readFilesCommand zcat <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>--soloFeatures GeneFull <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>--soloMultiMappers EM <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>--clipAdapterType CellRanger4 <span class="at">--outFilterScoreMin</span> 30 <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>--soloCBwhitelist 3M-february-2018.txt <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>--soloType CB_UMI_Simple <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>--soloCBlen 16 <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>--soloUMIstart 17 <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>--soloUMIlen 12 <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>--soloBarcodeReadLength 0 <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>--soloCellFilter EmptyDrops_CR <span class="dt">\</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>--limitOutSJcollapsed 5000000 <span class="dt">\</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>--runThreadN 8</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /home/ucbtvmr/Scratch/4_scRNAseq/preprocessing/star_2.7.10b/ensembl107/Solo.out/GeneFull</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> raw_EM</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> raw_EM</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="co"># move multimapped-containing matrix to /raw_EM</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> ../raw/UniqueAndMult-EM.mtx .</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../raw/features.tsv .</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../raw/barcodes.tsv .</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co"># rename multimapped-containing matrix to 'matrix.mtx' for --soloCellFiltering</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> UniqueAndMult-EM.mtx matrix.mtx</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Change working directory to the one containing Aligned.out.sam</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /home/ucbtvmr/Scratch/4_scRNAseq/preprocessing/star_2.7.10b/ensembl107</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Call cells</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a><span class="ex">STAR</span> <span class="at">--runMode</span> soloCellFiltering ./Solo.out/GeneFull/raw_EM ./Solo.out/GeneFull/filtered_EM/filtered_EM <span class="at">--soloCellFilter</span> EmptyDrops_CR</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ul></li>
</ul>
</section>
</section>
<section id="scrna-seq-analysis-in-seurat" class="level2">
<h2 class="anchored" data-anchor-id="scrna-seq-analysis-in-seurat">scRNA-seq analysis in <code>Seurat</code></h2>
<section id="load-cell-x-gene-matrices" class="level3">
<h3 class="anchored" data-anchor-id="load-cell-x-gene-matrices">Load cell x gene matrices</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>path1 <span class="ot">&lt;-</span> <span class="st">"/home/chongmorrison/Dropbox/VCM-StarBook/ResearchProjects/parapineal/preprocessing/foxd3flhGFP/cellranger7.1.0_Lawson/filtered_feature_bc_matrix"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>path2 <span class="ot">&lt;-</span> <span class="st">"/home/chongmorrison/Dropbox/VCM-StarBook/ResearchProjects/parapineal/preprocessing/foxd3flhGFP/starsolo2.7.10b_ens107/filtered_EM"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Create <code>Seurat</code> objects and check nos. of cells and genes</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="fu">Read10X</span>(path1), <span class="at">project=</span><span class="st">"CellRanger-Lawson"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Feature names cannot have underscores ('_'), replacing with dashes
('-')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="fu">ReadSTARsolo</span>(path2), <span class="at">project=</span><span class="st">"STARsolo-ens107"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Feature names cannot have underscores ('_'), replacing with dashes
('-')</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 4770 cells x 36353 genes</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>(cr_law)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4770</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(cr_law)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 36353</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 4536 cells x 32522 genes</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>(ss_e107)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4536</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(ss_e107)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 32522</code></pre>
</div>
</div>
</section>
<section id="filter-genes" class="level3">
<h3 class="anchored" data-anchor-id="filter-genes">Filter genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract counts</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">LayerData</span>(cr_law, <span class="at">layer =</span> <span class="st">"counts"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Output a logical matrix specifying for each gene on whether or not there are more than zero counts per cell</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>nonzero <span class="ot">&lt;-</span> counts <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Sums all TRUE values and returns TRUE if more than 5 TRUE values per gene</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">rowSums</span>(nonzero) <span class="sc">&gt;=</span> <span class="dv">5</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Only keeping those genes expressed in more than 5 cells</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>filtered_counts <span class="ot">&lt;-</span> counts[keep_genes, ]</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign to Seurat object (make new SeuratObject)</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(filtered_counts, <span class="at">meta.data =</span> cr_law<span class="sc">@</span>meta.data)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 4770 cells x 24969 genes</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>(cr_law)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4770</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(cr_law)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 24969</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">LayerData</span>(ss_e107, <span class="at">layer =</span> <span class="st">"counts"</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>nonzero <span class="ot">&lt;-</span> counts <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">rowSums</span>(nonzero) <span class="sc">&gt;=</span> <span class="dv">5</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>filtered_counts <span class="ot">&lt;-</span> counts[keep_genes, ]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(filtered_counts, <span class="at">meta.data =</span> ss_e107<span class="sc">@</span>meta.data)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 4536 cells x 23946 genes</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>(ss_e107) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4536</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(ss_e107)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 23946</code></pre>
</div>
</div>
</section>
<section id="compute-additional-metrics---percent.mito-ribo-complexity" class="level3">
<h3 class="anchored" data-anchor-id="compute-additional-metrics---percent.mito-ribo-complexity">Compute additional metrics - percent.mito, ribo, complexity</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>cr_law<span class="sc">$</span>percent.mt <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(cr_law, <span class="at">pattern =</span> <span class="st">"^mt-"</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>cr_law<span class="sc">$</span>log10GenesPerUMI <span class="ot">&lt;-</span> <span class="fu">log10</span>(cr_law<span class="sc">$</span>nFeature_RNA) <span class="sc">/</span> <span class="fu">log10</span>(cr_law<span class="sc">$</span>nCount_RNA)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>cr_law<span class="sc">$</span>percent.ribo <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(cr_law, <span class="at">pattern =</span><span class="st">"^rps"</span>) <span class="sc">+</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">PercentageFeatureSet</span>(cr_law, <span class="at">pattern =</span><span class="st">"^rpl"</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>ss_e107<span class="sc">$</span>percent.mt <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(ss_e107, <span class="at">pattern =</span> <span class="st">"^mt-"</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>ss_e107<span class="sc">$</span>log10GenesPerUMI <span class="ot">&lt;-</span> <span class="fu">log10</span>(ss_e107<span class="sc">$</span>nFeature_RNA) <span class="sc">/</span> <span class="fu">log10</span>(ss_e107<span class="sc">$</span>nCount_RNA)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>ss_e107<span class="sc">$</span>percent.ribo <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(ss_e107, <span class="at">pattern =</span><span class="st">"^rps"</span>) <span class="sc">+</span> </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">PercentageFeatureSet</span>(ss_e107, <span class="at">pattern =</span><span class="st">"^rpl"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="compute-cell-cycle-scores" class="level3">
<h3 class="anchored" data-anchor-id="compute-cell-cycle-scores">Compute cell cycle scores</h3>
<section id="obtain-cell-cycle-genes" class="level4">
<h4 class="anchored" data-anchor-id="obtain-cell-cycle-genes">Obtain cell cycle genes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>cell_cycle_genes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">text =</span> RCurl<span class="sc">::</span><span class="fu">getURL</span>(<span class="st">"https://raw.githubusercontent.com/hbc/tinyatlas/master/cell_cycle/Danio_rerio.csv"</span>))</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co"># call in Ensembl annotations</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>annotations <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'./ensembl_annotations.csv'</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># make a dataframe of cell cycle markers</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>cell_cycle_markers <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(cell_cycle_genes, annotations, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"geneID"</span> <span class="ot">=</span> <span class="st">"gene_id"</span>))</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>s_genes <span class="ot">&lt;-</span> cell_cycle_markers <span class="sc">%&gt;%</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(phase <span class="sc">==</span> <span class="st">"S"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="st">"gene_name"</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>g2m_genes <span class="ot">&lt;-</span> cell_cycle_markers <span class="sc">%&gt;%</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(phase <span class="sc">==</span> <span class="st">"G2/M"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="st">"gene_name"</span>)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove non-zebrafish "CENPF" and "CKS2" from g2m_genes</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>g2m_genes <span class="ot">&lt;-</span> g2m_genes[<span class="sc">!</span>g2m_genes <span class="sc">==</span> <span class="fu">c</span>(<span class="st">"CENPF"</span>,<span class="st">"CKS2"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in g2m_genes == c("CENPF", "CKS2"): longer object length is not a
multiple of shorter object length</code></pre>
</div>
</div>
</section>
<section id="cell-cycle-scoring" class="level4">
<h4 class="anchored" data-anchor-id="cell-cycle-scoring">Cell cycle scoring</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalise counts for cell cycle scoring</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(cr_law)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing layer: counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(ss_e107)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing layer: counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cell cycle scoring</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(cr_law, <span class="at">g2m.features =</span> g2m_genes, <span class="at">s.features =</span> s_genes)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(ss_e107, <span class="at">g2m.features =</span> g2m_genes, <span class="at">s.features =</span> s_genes)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Add difference between G2M and S phase scores</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>cr_law<span class="sc">$</span>CC.Difference <span class="ot">&lt;-</span> cr_law<span class="sc">$</span>S.Score <span class="sc">-</span> cr_law<span class="sc">$</span>G2M.Score</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>ss_e107<span class="sc">$</span>CC.Difference <span class="ot">&lt;-</span> ss_e107<span class="sc">$</span>S.Score <span class="sc">-</span> ss_e107<span class="sc">$</span>G2M.Score</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="cell-filtering" class="level3">
<h3 class="anchored" data-anchor-id="cell-filtering">Cell filtering?</h3>
<p>In a lot of cases, studies, example workflows etc. ‘poor’ quality cells are filtered out based on arbitrary metrics e.g.&nbsp;number of transcripts/genes suggestive of multiplets, high percentage mitochondrial transcripts suggestive of dying cells.</p>
<p>I judge these to be sound reasoning points…if you truly know the cellular composition of the dataset. Unlike gene-level filtering where you only make an <em>a priori</em> definition of an ‘expressed’ gene i.e.&nbsp;in at least X number of cells, cell filtering based on biological conditions dictated by the metrics mentioned above is more nuanced and (I believe) could introduce unnecessary bias and/or risk losing ‘good’ cells.</p>
<p>Therefore, for this novel dataset, I take a quasi-unbiased approach :</p>
<ul>
<li><p>Check that *most* cells are largely healthy i.e.&nbsp;percent.mt &lt; 5</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoPlot</span>(cr_law, <span class="st">"percent.mt"</span>, <span class="at">group.by =</span> <span class="st">"Phase"</span>,</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">vlnplot.lineweight =</span> <span class="fl">0.2</span>, <span class="at">jitter.size =</span> <span class="fl">0.3</span>,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">main =</span> <span class="st">"PercentageFeatureSet(^mt-), cr_law"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoPlot</span>(ss_e107, <span class="st">"percent.mt"</span>, <span class="at">group.by =</span> <span class="st">"Phase"</span>,</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">vlnplot.lineweight =</span> <span class="fl">0.2</span>, <span class="at">jitter.size =</span> <span class="fl">0.3</span>,</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">main =</span> <span class="st">"PercentageFeatureSet(^mt-), ss_e107"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div></li>
<li><p>Perform dimensional reduction i.e.&nbsp;clustering on all cells, then</p>
<ul>
<li><p>Check that potential multiplets are not forming spurious clusters, and are distributed across all called clusters</p></li>
<li><p>Also check that other ‘uninteresting’ metrics are not forming spurious clusters</p></li>
</ul></li>
</ul>
</section>
</section>
<section id="dimensional-reductionclustering" class="level2">
<h2 class="anchored" data-anchor-id="dimensional-reductionclustering">Dimensional reduction/clustering</h2>
<section id="sctransform-normalise-find-variable-genes-regress-unwanted-variation" class="level3">
<h3 class="anchored" data-anchor-id="sctransform-normalise-find-variable-genes-regress-unwanted-variation">SCTransform (‘normalise, find variable genes, regress unwanted variation’)</h3>
<p>I previously explored the data to check if there was significant variation between the cell types driven by ribosomal/mitochondrial content. <strong>Supplementary Section 1 (in progress).</strong></p>
<p>There was clear variation being driven by cell cycle phase. This was expected given a neuronal-enriched population at this developmental stage.</p>
<p>Altogether these inform the <code>SCTransform</code> parameters, where <code>CC.Difference</code> is used instead of <code>Phase</code> to maintain the separation between progenitors and non-cycling cells :</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(cr_law) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">SCTransform</span>(cr_law,</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">vars.to.regress =</span> <span class="fu">c</span>(<span class="st">"percent.ribo"</span>, <span class="st">"percent.mt"</span>, <span class="st">"CC.Difference"</span>),</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">vst.flavor=</span><span class="st">"v2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Running SCTransform on assay: RNA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Running SCTransform on layer: counts</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>vst.flavor='v2' set. Using model with fixed slope and excluding poisson genes.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Variance stabilizing transformation of count matrix of size 24969 by 4770</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Model formula is y ~ log_umi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Get Negative Binomial regression parameters per gene</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using 2000 genes, 4770 cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 16 outliers - those will be ignored in fitting/regularization step</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Second step: Get residuals using fitted parameters for 24969 genes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing corrected count matrix for 24969 genes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating gene attributes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Wall clock passed: Time difference of 15.94846 secs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Determine variable features</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Regressing out percent.ribo, percent.mt, CC.Difference</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering data matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Set default assay to SCT</code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(ss_e107) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">SCTransform</span>(ss_e107,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">vars.to.regress =</span> <span class="fu">c</span>(<span class="st">"percent.ribo"</span>, <span class="st">"percent.mt"</span>, <span class="st">"CC.Difference"</span>),</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">vst.flavor=</span><span class="st">"v2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Running SCTransform on assay: RNA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Running SCTransform on layer: counts</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>vst.flavor='v2' set. Using model with fixed slope and excluding poisson genes.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Variance stabilizing transformation of count matrix of size 23946 by 4536</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Model formula is y ~ log_umi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Get Negative Binomial regression parameters per gene</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using 2000 genes, 4536 cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 20 outliers - those will be ignored in fitting/regularization step</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Second step: Get residuals using fitted parameters for 23946 genes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing corrected count matrix for 23946 genes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating gene attributes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Wall clock passed: Time difference of 13.02944 secs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Determine variable features</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Regressing out percent.ribo, percent.mt, CC.Difference</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering data matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Set default assay to SCT</code></pre>
</div>
</div>
</section>
<section id="unsupervised-clustering" class="level3">
<h3 class="anchored" data-anchor-id="unsupervised-clustering">‘Unsupervised’ Clustering</h3>
<p>As clustering is more an art than exact science, I empirically determined the main parameters - <code>algorithm</code>, <code>npcs</code>, <code>k.param</code>, and <code>resolution</code> - and compared the result between the two pre-processing workflows/<code>Seurat</code> objects.</p>
<p><strong>‘Picking’ the number of Principal Component dimensions <code>npcs</code></strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(cr_law, <span class="at">npcs =</span> <span class="dv">60</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  sncb, stmn1b, tuba1c, ckbb, gap43, nrxn2a, elavl4, nrxn1a, rbfox1, onecut1 
       stx1b, tmsb, ywhag2, gng3, TENM2, elavl3, BX548062.1, GFP, robo2, snap25a 
       stmn2a, c1ql4a, celf4, id4, lhx9, raly, vamp2, mllt11, atp6v0cb, stmn4 
Negative:  si:ch211-263k4.2, gli2a, tns1b, jarid2b, lmx1bb, pard3bb, dock1, foxp4, zbtb16a, ctdp1 
       znrf3, ror1, cachd1, pax5, pdzrn3b, cdon, tenm4, lama5, rxraa, gli3 
       BX571942.1, diaph3, zeb2a, BX005454.1, trps1, slc35f1, dacha, nxn, hs6st1a, si:dkey-112m2.1 
PC_ 2 
Positive:  pax5, efna2a, tenm4, FP016248.1, ncam1a, her4.2, pax2b, lama5, dacha, nkain2 
       LOC100537384, lmx1bb, en1b, efnb3b, bcam, en2b, her4.1, pax3a, unc5c, si:ch211-57n23.4 
       elavl3, BX571942.1, ctnna2, pax2a, nova2, sema5ba, her15.1, LOC100535496, znf423, CR847968.1 
Negative:  alx4a, lrmda, mcamb, trpm1b, ltk, nrp2a, alx4b, adam19a, rxraa, pnp4a 
       aqp1a.1, gpnmb, si:ch211-243a20.3, sox5, sox10, impdh1b, cdh11, pcbp4, kank3, ttyh2l 
       gab2, pdzrn3b, anks1b, atp11a, apoda.1, defbl1, gmps, paics, sox6, pcdh9 
PC_ 3 
Positive:  CACNA2D1, sez6b, ptprfb, efna3a, znf804a, lrfn5a, lrp1bb, cacna1ha, BX548062.1, grm8a 
       mdga1, unc5a, znf536, TENM2, cntn3a.2, GRIK2, raly, grik4, cpne5b, nlgn1 
       nrxn2a, b4galnt4a, rbfox3b, kirrel3a, nav3, robo2, rbfox1, dscamb, GALNTL6, igsf21a 
Negative:  sncb, islr2, gap43, tuba1c, gng3, mllt11, stmn1b, atp6v0cb, tmsb2, ywhag2 
       cplx2l, stmn2a, zgc:65894, cotl1, vamp2, rtn1b, gapdhs, inab, tuba1a, rtn1a 
       bsg, tmsb, eno1a, tpi1b, map1ab, aldocb, pou3f1, elavl4, rnasekb, tac1 
PC_ 4 
Positive:  lhx9, raly, mdga1, znf804a, r3hdm1, rbfox3b, nr2f2, sez6b, ckbb, onecut1 
       GRIK2, chrm4a, lrfn5a, lhx2b, cpne7, cacna1ha, pdzrn4, grin2aa, evx2, GFP 
       atp2b1a, igsf21a, CACNA2D1, mab21l1, cntn3a.2, si:ch211-215d8.2, caln2, LOC110439372, grm8a, samd10b 
Negative:  celf5a, si:dkeyp-14d3.1, epha4b, thsd7ba, islr2, elavl4, pcdh9, cacna1i, edil3a, nlgn3a 
       msi2b, ntm, syt1a, rnf220a, neto1l, sema6bb, cadm1a, LOC100536716, CR847968.1, NPAS3 
       fam19a5a, enox2, ptprga, cntn4, brinp1, si:dkey-35m8.1, rab6ba, meis2a, plxna4, sox6 
PC_ 5 
Positive:  elavl3, pax5, pax7a, ctnna2, nkain2, efna2a, si:ch211-57n23.4, plxna2, pax2b, dlb 
       her4.2, en1b, thsd7aa, ebf3a, CR847968.1, nhlh2, en2b, pax2a, ebf2, srrm4 
       FP016248.1, hes6, LOC100537384, insm1b, sox11b, pdlim5a, XLOC-003692, si:ch73-386h18.1, myt1b, LOC101882472 
Negative:  lmx1bb, si:ch211-263k4.2, si:dkey-112m2.1, fabp11a, rspo2, ptprub, si:ch211-285f17.1, BX005454.1, brinp2, slit3 
       cdon, zgc:162939, CR848683.2, wnt4a, cyp26c1, rspo3, lrp2a, ntm, epha7, kirrel3a 
       lzts2a, gli2a, wnt3a, cachd1, unc5b, pax6a, unc5a, tnfrsf19, pawr, cldn5a </code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(cr_law, <span class="at">ndims=</span><span class="dv">60</span>, <span class="at">reduction=</span><span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(ss_e107, <span class="at">npcs =</span> <span class="dv">60</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  gli2a, jarid2b, si:ch211-263k4.2, lmx1bb, pard3bb, dock1, zbtb16a, ctdp1, foxp4, znrf3 
       CT027638.1, cachd1, ror1, cdon, pdzrn3b, tenm4, qkib, BX005454.1, pax5, gli3 
       BX571942.1, trps1, lama5, dacha, zeb2a, rxraa, efnb3b, nxn, hs6st1a, fbxw7 
Negative:  sncb, ckbb, tuba1c, gap43, elavl4, nrxn1a, nrxn2a, rbfox1, tmsb, stx1b 
       paqr7b, ywhag2, gng3, GFP, tuba1a, TENM2, snap25a, onecut1, stmn2a, rtn1a 
       elavl3, tubb5, id4, mllt11, atp6v0cb, lhx9, stmn1b, raly, cadm4, robo2.1 
PC_ 2 
Positive:  alx4a, lrmda, trpm1b, mcamb, nrp2a, ltk, alx4b, pnp4a, aqp1a.1, gpnmb 
       si:ch211-243a20.3, rxraa, impdh1b, sox10, sox5, cdh11, ttyh2l, kank3, rab32a, gab2 
       apoda.1, defbl1, gmps, pcbp4, paics, atp11a, pdzrn3b, sox6, ldhba, atic 
Negative:  pax5, efna2a, ncam1a, tenm4, FP016248.1, dacha, her4.2, nkain2, ctnna2, sema5ba 
       lama5, pax2b, lmx1bb, ENSDARG00000059707, efnb3b, adgrb3, bcam, BX571942.1, pax3a, en1b 
       BX465834.1, pou2f2a.1, her4.1, BX005454.1, en2b, brsk2b, nova2, cdon, BX663520.2, pax2a 
PC_ 3 
Positive:  CACNA2D1, efna3a, TENM2, ptprfb, sez6b, znf536, rbms3, znf804a, unc5a, nrxn1a 
       nrxn2a, cacna1ha, lrfn5a, grm8a, dscamb, b4galnt4a, cpne5b, chl1b, kcnma1a, mdga1 
       gria3b, cadm4, nrg3b, nlgn1, meis1b, GRIK2, raly, rbfox1, SRCIN1, LRFN2 
Negative:  tuba1c, sncb, gap43, gng3, tuba1a, mllt11, ywhag2, atp6v0cb, tmsb, paqr7b 
       tmsb2, zgc:65894, stmn2a, islr2, rtn1a, si:dkey-276j7.1, ckbb, vamp2, stmn1b, cplx2l 
       bsg, cotl1, rtn1b, tubb5, gapdhs, rnasekb, hmgb3a, zgc:153426, pou3f1, tpi1b 
PC_ 4 
Positive:  raly, sez6b, mdga1, znf804a, GRIK2, lhx9, cacna1ha, lrfn5a, grin2aa, rbfox3b 
       CACNA2D1, cpne7, r3hdm1, igsf21a, chrm4a, grm8a, nr2f2, pdzrn4, LRFN2, samd10b 
       caln2, evx2, cntn3a.2, robo2.1, atp2b1a, lhx2b, rbfox1, GALNTL6, si:ch211-215d8.2, cpne5b 
Negative:  celf5a, TMEM132C, islr2, msi2b, elavl4, thsd7ba, epha4b, cadm1a, CR847968.1, edil3a 
       ntm, rnf220a, cacna1ia, nlgn3a, sox6, syt1a, neto1l, enox2, ebf3a, tafa5a 
       map1ab, plxna3, khdrbs2, ptprga, meis2a.1, snap25b, atp1b1b, plxna4, isl1, pcp4a 
PC_ 5 
Positive:  elavl3, pax7a, pax5, efna2a, nkain2, ctnna2, her4.2, dlb, pax2b, en1b 
       ebf3a.1, gadd45gb.1, nhlh2, plxna2, pax2a, en2b, thsd7aa, sox11b, ebf2, srrm4 
       FP016248.1, hes6, CR847968.1, si:ch73-386h18.1, insm1b, myt1b, her4.2.1, abhd6a, scrt2, pou2f2a.1 
Negative:  lmx1bb, TMEM132C.1, fabp11a, si:ch211-263k4.2, rspo2, BX005454.1, ptprub, si:ch211-285f17.1, brinp2, slit3 
       cdon, aopep, CR848683.2, wnt4, ntm, rspo3, cyp26c1, gli2a, lrp2a, epha7 
       kirrel3a, wnt3a, cachd1, unc5a, unc5b, pax6a, pax6b, pawr, tnfrsf19, barhl2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(ss_e107, <span class="at">ndims=</span><span class="dv">60</span>, <span class="at">reduction=</span><span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Based on the elbow plots above I went with <code>npcs = 40</code>. <code>k.param</code> is kept at the default <code>20</code>, and the Leiden <code>algorithm = 4</code> is selected.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(cr_law, <span class="at">npcs =</span> <span class="dv">40</span>) <span class="sc">%&gt;%</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FindNeighbors</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>) <span class="sc">%&gt;%</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FindClusters</span>(<span class="at">algorithm =</span> <span class="dv">4</span>, <span class="at">resolution =</span> <span class="fu">c</span>(<span class="fl">1.0</span>), <span class="at">method=</span><span class="st">"igraph"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>, <span class="at">n.neighbors =</span> <span class="dv">20</span>) <span class="co"># to match default k.param=20</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  sncb, stmn1b, tuba1c, ckbb, gap43, nrxn2a, elavl4, nrxn1a, rbfox1, onecut1 
       stx1b, tmsb, ywhag2, gng3, TENM2, elavl3, BX548062.1, GFP, robo2, snap25a 
       stmn2a, c1ql4a, celf4, id4, lhx9, raly, vamp2, mllt11, atp6v0cb, stmn4 
Negative:  si:ch211-263k4.2, gli2a, tns1b, jarid2b, lmx1bb, pard3bb, dock1, foxp4, zbtb16a, ctdp1 
       znrf3, ror1, cachd1, pax5, pdzrn3b, cdon, tenm4, lama5, rxraa, gli3 
       BX571942.1, diaph3, zeb2a, BX005454.1, trps1, slc35f1, dacha, nxn, hs6st1a, si:dkey-112m2.1 
PC_ 2 
Positive:  pax5, efna2a, tenm4, FP016248.1, ncam1a, her4.2, pax2b, lama5, dacha, nkain2 
       LOC100537384, lmx1bb, en1b, efnb3b, bcam, en2b, her4.1, pax3a, unc5c, si:ch211-57n23.4 
       elavl3, BX571942.1, ctnna2, pax2a, nova2, sema5ba, her15.1, LOC100535496, znf423, CR847968.1 
Negative:  alx4a, lrmda, mcamb, trpm1b, ltk, nrp2a, alx4b, adam19a, rxraa, pnp4a 
       aqp1a.1, gpnmb, si:ch211-243a20.3, sox5, sox10, impdh1b, cdh11, pcbp4, kank3, ttyh2l 
       gab2, pdzrn3b, anks1b, atp11a, apoda.1, defbl1, gmps, paics, sox6, pcdh9 
PC_ 3 
Positive:  CACNA2D1, sez6b, ptprfb, efna3a, znf804a, lrfn5a, lrp1bb, cacna1ha, BX548062.1, grm8a 
       mdga1, unc5a, znf536, TENM2, cntn3a.2, GRIK2, raly, grik4, cpne5b, nlgn1 
       nrxn2a, b4galnt4a, rbfox3b, kirrel3a, nav3, robo2, rbfox1, dscamb, GALNTL6, igsf21a 
Negative:  sncb, islr2, gap43, tuba1c, gng3, mllt11, stmn1b, atp6v0cb, tmsb2, ywhag2 
       cplx2l, stmn2a, zgc:65894, cotl1, vamp2, rtn1b, gapdhs, inab, tuba1a, rtn1a 
       bsg, tmsb, eno1a, tpi1b, map1ab, aldocb, pou3f1, elavl4, rnasekb, tac1 
PC_ 4 
Positive:  celf5a, si:dkeyp-14d3.1, epha4b, thsd7ba, islr2, elavl4, pcdh9, cacna1i, edil3a, nlgn3a 
       msi2b, ntm, syt1a, rnf220a, neto1l, sema6bb, cadm1a, LOC100536716, CR847968.1, NPAS3 
       fam19a5a, enox2, ptprga, cntn4, brinp1, si:dkey-35m8.1, rab6ba, meis2a, plxna4, sox6 
Negative:  lhx9, raly, mdga1, znf804a, r3hdm1, rbfox3b, nr2f2, sez6b, ckbb, onecut1 
       GRIK2, chrm4a, lrfn5a, lhx2b, cpne7, cacna1ha, pdzrn4, grin2aa, evx2, GFP 
       atp2b1a, igsf21a, CACNA2D1, mab21l1, cntn3a.2, si:ch211-215d8.2, caln2, LOC110439372, grm8a, samd10b 
PC_ 5 
Positive:  elavl3, pax5, pax7a, ctnna2, nkain2, efna2a, si:ch211-57n23.4, plxna2, pax2b, dlb 
       her4.2, en1b, thsd7aa, ebf3a, CR847968.1, nhlh2, en2b, pax2a, ebf2, srrm4 
       FP016248.1, hes6, LOC100537384, insm1b, sox11b, pdlim5a, XLOC-003692, si:ch73-386h18.1, myt1b, LOC101882472 
Negative:  lmx1bb, si:ch211-263k4.2, si:dkey-112m2.1, fabp11a, rspo2, ptprub, si:ch211-285f17.1, BX005454.1, brinp2, slit3 
       cdon, zgc:162939, CR848683.2, wnt4a, cyp26c1, rspo3, lrp2a, ntm, epha7, kirrel3a 
       lzts2a, gli2a, wnt3a, cachd1, unc5b, pax6a, unc5a, tnfrsf19, pawr, cldn5a </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing SNN</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>20:09:52 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>20:09:52 Read 4770 rows and found 40 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>20:09:52 Using Annoy for neighbor search, n_neighbors = 20</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>20:09:52 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
20:09:52 Writing NN index file to temp file /tmp/RtmpvEtMAu/file39b670137b6b
20:09:52 Searching Annoy index using 1 thread, search_k = 2000
20:09:53 Annoy recall = 100%
20:09:53 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 20
20:09:54 Initializing from normalized Laplacian + noise (using RSpectra)
20:09:54 Commencing optimization for 500 epochs, with 130740 positive edges
20:09:57 Optimization finished</code></pre>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(ss_e107, <span class="at">npcs =</span> <span class="dv">40</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FindNeighbors</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FindClusters</span>(<span class="at">algorithm =</span> <span class="dv">4</span>, <span class="at">resolution =</span> <span class="fu">c</span>(<span class="fl">1.0</span>), <span class="at">method=</span><span class="st">"igraph"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RunUMAP</span>(<span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>, <span class="at">n.neighbors =</span> <span class="dv">20</span>) <span class="co"># to match default k.param=20</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  gli2a, jarid2b, si:ch211-263k4.2, lmx1bb, pard3bb, dock1, zbtb16a, ctdp1, foxp4, znrf3 
       CT027638.1, cachd1, ror1, cdon, pdzrn3b, tenm4, qkib, BX005454.1, pax5, gli3 
       BX571942.1, trps1, lama5, dacha, zeb2a, rxraa, efnb3b, nxn, hs6st1a, fbxw7 
Negative:  sncb, ckbb, tuba1c, gap43, elavl4, nrxn1a, nrxn2a, rbfox1, tmsb, stx1b 
       paqr7b, ywhag2, gng3, GFP, tuba1a, TENM2, snap25a, onecut1, stmn2a, rtn1a 
       elavl3, tubb5, id4, mllt11, atp6v0cb, lhx9, stmn1b, raly, cadm4, robo2.1 
PC_ 2 
Positive:  alx4a, lrmda, trpm1b, mcamb, nrp2a, ltk, alx4b, pnp4a, aqp1a.1, gpnmb 
       si:ch211-243a20.3, rxraa, impdh1b, sox10, sox5, cdh11, ttyh2l, kank3, rab32a, gab2 
       apoda.1, defbl1, gmps, pcbp4, paics, atp11a, pdzrn3b, sox6, ldhba, atic 
Negative:  pax5, efna2a, ncam1a, tenm4, FP016248.1, dacha, her4.2, nkain2, ctnna2, sema5ba 
       lama5, pax2b, lmx1bb, ENSDARG00000059707, efnb3b, adgrb3, bcam, BX571942.1, pax3a, en1b 
       BX465834.1, pou2f2a.1, her4.1, BX005454.1, en2b, brsk2b, nova2, cdon, BX663520.2, pax2a 
PC_ 3 
Positive:  CACNA2D1, efna3a, TENM2, ptprfb, sez6b, znf536, rbms3, znf804a, unc5a, nrxn1a 
       nrxn2a, cacna1ha, lrfn5a, grm8a, dscamb, b4galnt4a, cpne5b, chl1b, kcnma1a, mdga1 
       gria3b, cadm4, nrg3b, nlgn1, meis1b, GRIK2, raly, rbfox1, SRCIN1, LRFN2 
Negative:  tuba1c, sncb, gap43, gng3, tuba1a, mllt11, ywhag2, atp6v0cb, tmsb, paqr7b 
       tmsb2, zgc:65894, stmn2a, islr2, rtn1a, si:dkey-276j7.1, ckbb, vamp2, stmn1b, cplx2l 
       bsg, cotl1, rtn1b, tubb5, gapdhs, rnasekb, hmgb3a, zgc:153426, pou3f1, tpi1b 
PC_ 4 
Positive:  celf5a, TMEM132C, islr2, msi2b, elavl4, thsd7ba, epha4b, cadm1a, CR847968.1, edil3a 
       ntm, rnf220a, cacna1ia, nlgn3a, sox6, syt1a, neto1l, enox2, ebf3a, tafa5a 
       map1ab, plxna3, khdrbs2, ptprga, meis2a.1, snap25b, atp1b1b, plxna4, isl1, pcp4a 
Negative:  raly, sez6b, mdga1, znf804a, GRIK2, lhx9, cacna1ha, lrfn5a, grin2aa, rbfox3b 
       CACNA2D1, cpne7, r3hdm1, igsf21a, chrm4a, grm8a, nr2f2, pdzrn4, LRFN2, samd10b 
       caln2, evx2, cntn3a.2, robo2.1, atp2b1a, lhx2b, rbfox1, GALNTL6, si:ch211-215d8.2, cpne5b 
PC_ 5 
Positive:  lmx1bb, TMEM132C.1, fabp11a, si:ch211-263k4.2, rspo2, BX005454.1, ptprub, si:ch211-285f17.1, brinp2, slit3 
       cdon, aopep, CR848683.2, wnt4, ntm, rspo3, cyp26c1, gli2a, lrp2a, epha7 
       kirrel3a, wnt3a, cachd1, unc5a, unc5b, pax6a, pax6b, pawr, tnfrsf19, barhl2 
Negative:  elavl3, pax7a, pax5, efna2a, nkain2, ctnna2, her4.2, dlb, pax2b, en1b 
       ebf3a.1, gadd45gb.1, nhlh2, plxna2, pax2a, en2b, thsd7aa, sox11b, ebf2, srrm4 
       FP016248.1, hes6, CR847968.1, si:ch73-386h18.1, insm1b, myt1b, her4.2.1, abhd6a, scrt2, pou2f2a.1 
Computing nearest neighbor graph
Computing SNN
20:10:00 UMAP embedding parameters a = 0.9922 b = 1.112
20:10:00 Read 4536 rows and found 40 numeric columns
20:10:00 Using Annoy for neighbor search, n_neighbors = 20
20:10:00 Building Annoy index with metric = cosine, n_trees = 50
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
20:10:00 Writing NN index file to temp file /tmp/RtmpvEtMAu/file39b67c778279
20:10:00 Searching Annoy index using 1 thread, search_k = 2000
20:10:01 Annoy recall = 100%
20:10:01 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 20
20:10:02 Initializing from normalized Laplacian + noise (using RSpectra)
20:10:02 Commencing optimization for 500 epochs, with 122026 positive edges
20:10:05 Optimization finished</code></pre>
</div>
</div>
<p><strong>Save <code>Seurat</code> objects to load later without re-running all the above:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set DefaultAssay (and slot) and save objects for visualisation</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(cr_law) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>cr_law <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(cr_law)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing layer: counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(ss_e107) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>ss_e107 <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(ss_e107)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing layer: counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(cr_law, <span class="at">file=</span><span class="st">"./foxd3flh-5k_cr-law.RData"</span>)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ss_e107, <span class="at">file=</span><span class="st">"./foxd3flh-5k_ss-e107.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="visualise-and-explore-clustering-results-with-dittoseq" class="level2">
<h2 class="anchored" data-anchor-id="visualise-and-explore-clustering-results-with-dittoseq">Visualise and explore clustering results with <code>dittoSeq</code></h2>
<section id="umap-reduction" class="level3">
<h3 class="anchored" data-anchor-id="umap-reduction">UMAP reduction</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(cr_law, <span class="st">"SCT_snn_res.1"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"CellRanger7.1.0"</span>,</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"Lawson2020"</span>,</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"clusters"</span>,</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">do.label=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(ss_e107, <span class="st">"SCT_snn_res.1"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"STARsolo2.7.10b"</span>,</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"ensembl107"</span>,</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"clusters"</span>,</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">do.label=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="identify-epiphysis-clusters" class="level2">
<h2 class="anchored" data-anchor-id="identify-epiphysis-clusters">Identify epiphysis clusters</h2>
<p>The epiphysis shares several of its known i.e.&nbsp;well-characterised markers with other neuronal/cranial cell types. For example, <em>bsx</em> is also expressed in the hypothalamus, <em>noto</em> is also expressed in the telencephalon, <em>otx5</em> is also expressed in the retina. Inferring the epiphysis clusters from any of these genes alone is futile:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes that are well known to be expressed in the epiphysis</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"bsx"</span>, <span class="st">"noto"</span>, <span class="st">"otx5"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoPlotVarsAcrossGroups</span>(cr_law, markers, <span class="at">group.by =</span> <span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">vlnplot.lineweight =</span> <span class="fl">0.2</span>, <span class="at">jitter.size =</span> <span class="fl">0.3</span>,</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">main =</span> <span class="st">"cr_law (bsx/noto/otx5)"</span>, <span class="at">summary.fxn =</span> max, <span class="at">adjustment =</span> <span class="cn">NULL</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoPlotVarsAcrossGroups</span>(ss_e107, markers, <span class="at">group.by =</span> <span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">vlnplot.lineweight =</span> <span class="fl">0.2</span>, <span class="at">jitter.size =</span> <span class="fl">0.3</span>,</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">main =</span> <span class="st">"ss_e107 (bsx/noto/otx5)"</span>, <span class="at">summary.fxn =</span> max, <span class="at">adjustment =</span> <span class="cn">NULL</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I reasoned that, by using instead the <strong>mean</strong> expression values of these three genes combined, one could eliminate clusters that are least likely to be epiphysis cells.</p>
<p>For example, an epiphysis cell is likely to co-express bsx (1) + noto (1) + otx5 (1) = 3. Mean expression value = 3 / 3 = 1.</p>
<p>On the other hand, a retina cell would not express bsx ( &lt; 1) + noto (&lt; 1), but express + otx5 (1) = &lt; 3. Mean expression value = &lt;3 / 3 = &lt; 1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoPlotVarsAcrossGroups</span>(cr_law, markers, <span class="at">group.by =</span> <span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">vlnplot.lineweight =</span> <span class="fl">0.2</span>, <span class="at">jitter.size =</span> <span class="fl">0.3</span>,</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">main =</span> <span class="st">"cr_law (mean of bsx + noto + otx5)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoPlotVarsAcrossGroups</span>(ss_e107, markers, <span class="at">group.by =</span> <span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">vlnplot.lineweight =</span> <span class="fl">0.2</span>, <span class="at">jitter.size =</span> <span class="fl">0.3</span>,</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">main =</span> <span class="st">"ss_e107 (mean of bsx + noto + otx5)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>This worked quite well, and one can appreciate two clusters-of-interest likely to be epiphysis cells.</strong> Furthermore, this observation is robust and reproducible from cell x gene matrices generated using the two dissimilar pre-processing pipelines.</p>
<ul>
<li><p><code>cr_law</code> clusters-of-interest = 2 and 18</p></li>
<li><p><code>ss_e107</code> clusters-of-interest = 2 and 19</p></li>
</ul>
<p>Let’s now look at <code>sox1a</code> that is expressed in the parapineal as it emerges from the anterior pineal region (Lekk <em>et al</em>. 2019). The following plots show that <code>sox1a</code> expression can be detected in the clusters-of-interest.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">multi_dittoDimPlot</span>(cr_law, <span class="fu">c</span>(markers,<span class="st">"sox1a"</span>), <span class="at">order=</span><span class="st">"increasing"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">multi_dittoDimPlot</span>(ss_e107, <span class="fu">c</span>(markers,<span class="st">"sox1a"</span>), <span class="at">order=</span><span class="st">"increasing"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>As both matrices consistently produced similar results so far, I am proceeding further with <code>ss_e107</code> only.</p>
<p>How many cells (roughly) are present in the clusters-of-interest? ~250 cells in total.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoBarPlot</span>(ss_e107, <span class="st">"SCT_snn_res.1"</span>, <span class="at">group.by =</span> <span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">scale =</span> <span class="st">"count"</span>,</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">cells.use =</span> <span class="fu">meta</span>(<span class="st">"SCT_snn_res.1"</span>, ss_e107) <span class="sc">==</span> <span class="fu">c</span>(<span class="st">"2"</span>, <span class="st">"19"</span>),</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"No. of cells"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="sanity-checks-on-clustering" class="level2">
<h2 class="anchored" data-anchor-id="sanity-checks-on-clustering">Sanity checks on clustering</h2>
<p>Remember that I did not perform <a href="#cell-filtering">Cell filtering?</a> To evaluate if the clustering afterwards was largely driven by ‘true’ biological heterogeneity, in a qualitative sense:</p>
<section id="check-distribution-of-multiplets" class="level3">
<h3 class="anchored" data-anchor-id="check-distribution-of-multiplets">Check distribution of multiplets</h3>
<p>I run <code>scDblFinder</code> to identify multiplets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(<span class="fu">LayerData</span>(ss_e107, <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">layer=</span><span class="st">"counts"</span>))))</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">scDblFinder</span>(sce, <span class="at">clusters=</span><span class="fu">c</span>(<span class="fu">Idents</span>(ss_e107)),</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">BPPARAM=</span><span class="fu">MulticoreParam</span>(<span class="dv">2</span>, <span class="at">RNGseed=</span><span class="dv">1234</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>21 clusters</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Creating ~4410 artificial doublets...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dimensional reduction</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Evaluating kNN...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Training model...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>iter=0, 355 cells excluded from training.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>iter=1, 354 cells excluded from training.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>iter=2, 342 cells excluded from training.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Threshold found:0.497</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>369 (8.1%) doublets called</code></pre>
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(sce<span class="sc">$</span>scDblFinder.class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
singlet doublet 
   4167     369 </code></pre>
</div>
</div>
<p>Now port the results back to the original <code>ss_e107</code> object for visualisation. Called doublets are distributed randomly across entire dataset, which is ideal.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>ss_e107<span class="sc">$</span>scDblFinder.class <span class="ot">&lt;-</span> sce<span class="sc">$</span>scDblFinder.class</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(ss_e107, <span class="st">"scDblFinder.class"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"STARsolo2.7.10b_ensembl107"</span>,</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"class"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="what-about-other-metrics" class="level3">
<h3 class="anchored" data-anchor-id="what-about-other-metrics">What about other metrics?</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(ss_e107, <span class="st">"nCount_RNA"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"STARsolo2.7.10b_ensembl107"</span>,</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"nUMI"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(ss_e107, <span class="st">"nFeature_RNA"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"STARsolo2.7.10b_ensembl107"</span>,</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"nGenes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(ss_e107, <span class="st">"percent.ribo"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"STARsolo2.7.10b_ensembl107"</span>,</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"% ribosomal transcripts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(ss_e107, <span class="st">"percent.mt"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"STARsolo2.7.10b_ensembl107"</span>,</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"% mitochondrial transcripts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDimPlot</span>(ss_e107, <span class="st">"Phase"</span>, <span class="at">reduction.use=</span><span class="st">"umap"</span>,</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">main=</span><span class="st">"STARsolo2.7.10b_ensembl107"</span>,</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">sub=</span><span class="st">"SCT_snn_res.1"</span>,</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend.title=</span><span class="st">"cell cycle"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Aim1_part1_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The plots above suggests that the clusters-of-interest 2 and 19 vary from each other by cell cycle phase, and perhaps by subtle differences in the other metrics. This is something to bear in mind when interpreting results based on this clustering.</p>
<hr>
</section>
</section>
<section id="summary-notes-for-aim-1---part-1" class="level2">
<h2 class="anchored" data-anchor-id="summary-notes-for-aim-1---part-1">Summary notes for Aim 1 - Part 1</h2>
<ol type="1">
<li><p>Putative epiphysis cell clusters can be identified by scRNA-seq regardless of <em>sox1a</em> genotype.</p></li>
<li><p><em>Sox1a</em>-positive putative parapineal cells can be detected within these clusters.</p></li>
<li><p>Next step: <em>In vivo</em> validation of these putative clusters using imaging-based spatial transcriptomics (Vizgen MERSCOPE)</p></li>
</ol>
<hr>
</section>
<section id="references" class="level2">
<h2 class="anchored" data-anchor-id="references">References</h2>
<section id="packages" class="level3">
<h3 class="anchored" data-anchor-id="packages">Packages</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>pkgs <span class="ot">&lt;-</span> <span class="fu">cite_packages</span>(<span class="at">output =</span> <span class="st">"table"</span>, <span class="at">out.dir =</span> <span class="st">"."</span>)</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(pkgs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Package</th>
<th style="text-align: left;">Version</th>
<th style="text-align: left;">Citation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">base</td>
<td style="text-align: left;">4.4.2</td>
<td style="text-align: left;"><span class="citation" data-cites="base">R Core Team (<a href="#ref-base" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">BiocParallel</td>
<td style="text-align: left;">1.38.0</td>
<td style="text-align: left;"><span class="citation" data-cites="BiocPar">(<a href="#ref-BiocPar" role="doc-biblioref"><strong>BiocPar?</strong></a>)</span>….</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dittoSeq</td>
<td style="text-align: left;">1.16.0</td>
<td style="text-align: left;"><span class="citation" data-cites="dittoSeq">Bunis et al. (<a href="#ref-dittoSeq" role="doc-biblioref">2020</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">future</td>
<td style="text-align: left;">1.34.0</td>
<td style="text-align: left;"><span class="citation" data-cites="RJ-2021-048">Bengtsson (<a href="#ref-RJ-2021-048" role="doc-biblioref">2021</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">knitr</td>
<td style="text-align: left;">1.49</td>
<td style="text-align: left;"><span class="citation" data-cites="knitr20">(<a href="#ref-knitr20" role="doc-biblioref"><strong>knitr20?</strong></a>)</span>….</td>
</tr>
<tr class="even">
<td style="text-align: left;">Matrix</td>
<td style="text-align: left;">1.7.0</td>
<td style="text-align: left;"><span class="citation" data-cites="Matrix">Bates, Maechler, and Jagan (<a href="#ref-Matrix" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">RCurl</td>
<td style="text-align: left;">1.98.1.16</td>
<td style="text-align: left;"><span class="citation" data-cites="RCurl">Temple Lang (<a href="#ref-RCurl" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">reticulate</td>
<td style="text-align: left;">1.39.0</td>
<td style="text-align: left;"><span class="citation" data-cites="reticulate">Ushey, Allaire, and Tang (<a href="#ref-reticulate" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">rmarkdown</td>
<td style="text-align: left;">2.29</td>
<td style="text-align: left;"><span class="citation" data-cites="rmarkdo">(<a href="#ref-rmarkdo" role="doc-biblioref"><strong>rmarkdo?</strong></a>)</span>….</td>
</tr>
<tr class="even">
<td style="text-align: left;">scDblFinder</td>
<td style="text-align: left;">1.18.0</td>
<td style="text-align: left;"><span class="citation" data-cites="scDblFinder">Germain et al. (<a href="#ref-scDblFinder" role="doc-biblioref">2022</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Seurat</td>
<td style="text-align: left;">5.1.0</td>
<td style="text-align: left;"><span class="citation" data-cites="Seurat2">(<a href="#ref-Seurat2" role="doc-biblioref"><strong>Seurat2?</strong></a>)</span>….</td>
</tr>
<tr class="even">
<td style="text-align: left;">SingleCellExperiment</td>
<td style="text-align: left;">1.26.0</td>
<td style="text-align: left;"><span class="citation" data-cites="SingleC">(<a href="#ref-SingleC" role="doc-biblioref"><strong>SingleC?</strong></a>)</span>….</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tidyverse</td>
<td style="text-align: left;">2.0.0</td>
<td style="text-align: left;"><span class="citation" data-cites="tidyverse">Wickham et al. (<a href="#ref-tidyverse" role="doc-biblioref">2019</a>)</span></td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="previous-work" class="level3">
<h3 class="anchored" data-anchor-id="previous-work">Previous work</h3>
<ul>
<li>Mary Piper, Meeta Mistry, Jihe Liu, William Gammerdinger, &amp; Radhika Khetani. (2022, January 6). hbctraining/scRNA-seq_online: scRNA-seq Lessons from HCBC (first release). Zenodo. <a href="https://doi.org/10.5281/zenodo.5826256" class="uri">https://doi.org/10.5281/zenodo.5826256</a>.</li>
<li>I. Lekk, V. Duboc, A. Faro, S. Nicolaou, P. Blader and S.W. Wilson. <em>Sox1a mediates the ability of the parapineal to impart habenular left-right asymmetry</em>. eLife (2019), e47376, 8, <a href="https://doi.org/10.7554/eLife.47376" class="uri">https://doi.org/10.7554/eLife.47376</a></li>
</ul>
</section>
</section>
<section id="session-info" class="level2">
<h2 class="anchored" data-anchor-id="session-info">Session Info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.2 (2024-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/London
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] scDblFinder_1.18.0          grateful_0.2.10            
 [3] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
 [5] Biobase_2.64.0              GenomicRanges_1.56.2       
 [7] GenomeInfoDb_1.40.1         IRanges_2.38.1             
 [9] S4Vectors_0.42.1            BiocGenerics_0.50.0        
[11] MatrixGenerics_1.16.0       matrixStats_1.4.1          
[13] dittoSeq_1.16.0             lubridate_1.9.3            
[15] forcats_1.0.0               stringr_1.5.1              
[17] dplyr_1.1.4                 purrr_1.0.2                
[19] readr_2.1.5                 tidyr_1.3.1                
[21] tibble_3.2.1                ggplot2_3.5.1              
[23] tidyverse_2.0.0             Seurat_5.1.0               
[25] SeuratObject_5.0.2          sp_2.1-4                   
[27] reticulate_1.39.0           BiocParallel_1.38.0        
[29] future_1.34.0              

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22          splines_4.4.2            
  [3] later_1.3.2               BiocIO_1.14.0            
  [5] bitops_1.0-9              polyclip_1.10-7          
  [7] XML_3.99-0.17             fastDummies_1.7.4        
  [9] lifecycle_1.0.4           edgeR_4.2.2              
 [11] globals_0.16.3            lattice_0.22-6           
 [13] MASS_7.3-61               magrittr_2.0.3           
 [15] limma_3.60.6              plotly_4.10.4            
 [17] rmarkdown_2.29            yaml_2.3.10              
 [19] metapod_1.12.0            httpuv_1.6.15            
 [21] glmGamPoi_1.16.0          sctransform_0.4.1        
 [23] spam_2.11-0               spatstat.sparse_3.1-0    
 [25] cowplot_1.1.3             pbapply_1.7-2            
 [27] RColorBrewer_1.1-3        abind_1.4-8              
 [29] zlibbioc_1.50.0           Rtsne_0.17               
 [31] RCurl_1.98-1.16           rappdirs_0.3.3           
 [33] GenomeInfoDbData_1.2.12   ggrepel_0.9.6            
 [35] irlba_2.3.5.1             listenv_0.9.1            
 [37] spatstat.utils_3.1-1      pheatmap_1.0.12          
 [39] goftest_1.2-3             RSpectra_0.16-2          
 [41] dqrng_0.4.1               spatstat.random_3.3-2    
 [43] fitdistrplus_1.2-1        parallelly_1.39.0        
 [45] DelayedMatrixStats_1.26.0 leiden_0.4.3.1           
 [47] codetools_0.2-20          DelayedArray_0.30.1      
 [49] scuttle_1.14.0            tidyselect_1.2.1         
 [51] UCSC.utils_1.0.0          farver_2.1.2             
 [53] viridis_0.6.5             ScaledMatrix_1.12.0      
 [55] spatstat.explore_3.3-3    GenomicAlignments_1.40.0 
 [57] jsonlite_1.8.9            BiocNeighbors_1.22.0     
 [59] progressr_0.15.0          scater_1.32.1            
 [61] ggridges_0.5.6            survival_3.7-0           
 [63] tools_4.4.2               ica_1.0-3                
 [65] Rcpp_1.0.13-1             glue_1.8.0               
 [67] gridExtra_2.3             SparseArray_1.4.8        
 [69] xfun_0.49                 withr_3.0.2              
 [71] fastmap_1.2.0             bluster_1.14.0           
 [73] fansi_1.0.6               rsvd_1.0.5               
 [75] digest_0.6.37             timechange_0.3.0         
 [77] R6_2.5.1                  mime_0.12                
 [79] colorspace_2.1-1          scattermore_1.2          
 [81] tensor_1.5                spatstat.data_3.1-2      
 [83] utf8_1.2.4                generics_0.1.3           
 [85] renv_1.0.11               data.table_1.16.2        
 [87] rtracklayer_1.64.0        httr_1.4.7               
 [89] htmlwidgets_1.6.4         S4Arrays_1.4.1           
 [91] uwot_0.2.2                pkgconfig_2.0.3          
 [93] gtable_0.3.6              lmtest_0.9-40            
 [95] XVector_0.44.0            htmltools_0.5.8.1        
 [97] dotCall64_1.2             scales_1.3.0             
 [99] png_0.1-8                 spatstat.univar_3.1-1    
[101] scran_1.32.0              knitr_1.49               
[103] rstudioapi_0.17.1         rjson_0.2.23             
[105] tzdb_0.4.0                reshape2_1.4.4           
[107] curl_6.0.0                nlme_3.1-165             
[109] zoo_1.8-12                KernSmooth_2.23-24       
[111] vipor_0.4.7               parallel_4.4.2           
[113] miniUI_0.1.1.1            restfulr_0.0.15          
[115] pillar_1.9.0              grid_4.4.2               
[117] vctrs_0.6.5               RANN_2.6.2               
[119] promises_1.3.0            BiocSingular_1.20.0      
[121] beachmat_2.20.0           xtable_1.8-4             
[123] cluster_2.1.6             beeswarm_0.4.0           
[125] evaluate_1.0.1            locfit_1.5-9.10          
[127] Rsamtools_2.20.0          cli_3.6.3                
[129] compiler_4.4.2            rlang_1.1.4              
[131] crayon_1.5.3              future.apply_1.11.3      
[133] labeling_0.4.3            ggbeeswarm_0.7.2         
[135] plyr_1.8.9                stringi_1.8.4            
[137] viridisLite_0.4.2         deldir_2.0-4             
[139] Biostrings_2.72.1         munsell_0.5.1            
[141] lazyeval_0.2.2            spatstat.geom_3.3-3      
[143] Matrix_1.7-0              RcppHNSW_0.6.0           
[145] hms_1.1.3                 patchwork_1.3.0          
[147] sparseMatrixStats_1.16.0  statmod_1.5.0            
[149] shiny_1.9.1               ROCR_1.0-11              
[151] igraph_2.1.1              xgboost_1.7.8.1          </code></pre>
</div>
</div>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-Matrix" class="csl-entry" role="listitem">
Bates, Douglas, Martin Maechler, and Mikael Jagan. 2024. <em><span>Matrix</span>: Sparse and Dense Matrix Classes and Methods</em>. <a href="https://CRAN.R-project.org/package=Matrix">https://CRAN.R-project.org/package=Matrix</a>.
</div>
<div id="ref-RJ-2021-048" class="csl-entry" role="listitem">
Bengtsson, Henrik. 2021. <span>“A Unifying Framework for Parallel and Distributed Processing in r Using Futures.”</span> <em>The R Journal</em> 13 (2): 208–27. <a href="https://doi.org/10.32614/RJ-2021-048">https://doi.org/10.32614/RJ-2021-048</a>.
</div>
<div id="ref-dittoSeq" class="csl-entry" role="listitem">
Bunis, Daniel G., Jared Andrews, Gabriela K. Fragiadakis, Trevor D. Burt, and Marina Sirota. 2020. <span>“<span class="nocase">dittoSeq</span>: Universal User-Friendly Single-Cell and Bulk RNA Sequencing Visualization Toolkit.”</span> <em>Bioinformatics</em>, no. 22-23 (December): 5535–36. <a href="https://doi.org/10.1093/bioinformatics/btaa1011">https://doi.org/10.1093/bioinformatics/btaa1011</a>.
</div>
<div id="ref-scDblFinder" class="csl-entry" role="listitem">
Germain, Pierre-Luc, Aaron Lun, Carlos Garcia Meixide, Will Macnair, and Mark D. Robinson. 2022. <span>“Doublet Identification in Single-Cell Sequencing Data Using scDblFinder.”</span> <em>F1000research</em>. <a href="https://doi.org/10.12688/f1000research.73600.2">https://doi.org/10.12688/f1000research.73600.2</a>.
</div>
<div id="ref-base" class="csl-entry" role="listitem">
R Core Team. 2024. <em><span>R</span>: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-RCurl" class="csl-entry" role="listitem">
Temple Lang, Duncan. 2024. <em><span>RCurl</span>: General Network (HTTP/FTP/...) Client Interface for r</em>. <a href="https://CRAN.R-project.org/package=RCurl">https://CRAN.R-project.org/package=RCurl</a>.
</div>
<div id="ref-reticulate" class="csl-entry" role="listitem">
Ushey, Kevin, JJ Allaire, and Yuan Tang. 2024. <em><span class="nocase">reticulate</span>: Interface to <span>“<span>Python</span>”</span></em>. <a href="https://CRAN.R-project.org/package=reticulate">https://CRAN.R-project.org/package=reticulate</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span class="nocase">tidyverse</span>.”</span> <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">Copyright 2024, Vanessa Chong-Morrison</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>